From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 03:00:03 -0500
Subject: [PATCH] Simplify removeAll


diff --git a/src/main/java/io/papermc/paper/tag/BaseTag.java b/src/main/java/io/papermc/paper/tag/BaseTag.java
index 06e437b743e26afc1b4a1aab4113541f91c11bde..256600ce5b345d43bfdbe1f2a726e3512f874580 100644
--- a/src/main/java/io/papermc/paper/tag/BaseTag.java
+++ b/src/main/java/io/papermc/paper/tag/BaseTag.java
@@ -132,7 +132,7 @@ public abstract class BaseTag<T extends Keyed, C extends BaseTag<T, C>> implemen
     @NotNull
     public C not(@NotNull T...values) {
         this.checkLock();
-        this.tagged.removeAll(Lists.newArrayList(values));
+        Lists.newArrayList(values).forEach(this.tagged::remove);
         return (C) this;
     }
 
