From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 03:05:17 -0500
Subject: [PATCH] Use system array copy


diff --git a/src/main/java/org/bukkit/inventory/ShapedRecipe.java b/src/main/java/org/bukkit/inventory/ShapedRecipe.java
index 4eb31d5198adbd62644a53a32d5bf10ed001e802..0819a4886fb77215645e56aa75e9821381f4b6d9 100644
--- a/src/main/java/org/bukkit/inventory/ShapedRecipe.java
+++ b/src/main/java/org/bukkit/inventory/ShapedRecipe.java
@@ -75,9 +75,7 @@ public class ShapedRecipe implements Recipe, Keyed {
             lastLen = row.length();
         }
         this.rows = new String[shape.length];
-        for (int i = 0; i < shape.length; i++) {
-            this.rows[i] = shape[i];
-        }
+        System.arraycopy(shape, 0, this.rows, 0, shape.length);
 
         // Remove character mappings for characters that no longer exist in the shape
         HashMap<Character, RecipeChoice> newIngredients = new HashMap<>();
