From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 04:03:12 -0500
Subject: [PATCH] Use is empty


diff --git a/src/main/java/org/bukkit/plugin/SimplePluginManager.java b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
index 222f828268a94941be77aceb7bdcef8754f12351..f0cefde674d9f466c55b4b6d80189fd764c1fe4a 100644
--- a/src/main/java/org/bukkit/plugin/SimplePluginManager.java
+++ b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
@@ -121,7 +121,7 @@ public final class SimplePluginManager implements PluginManager {
 
         List<Plugin> result = new ArrayList<Plugin>();
 
-        if (!(server.getUpdateFolder().equals(""))) {
+        if (!(server.getUpdateFolder().isEmpty())) {
             updateDirectory = new File(directory, server.getUpdateFolder());
         }
 
