From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 04:04:10 -0500
Subject: [PATCH] Remove redundant checks


diff --git a/src/main/java/org/bukkit/material/MaterialData.java b/src/main/java/org/bukkit/material/MaterialData.java
index 524dd816ac169b1c2ad8e3a481cb28ec45bc1bbe..5f294024b958a4c32937f343ef1bb5b6c5003308 100644
--- a/src/main/java/org/bukkit/material/MaterialData.java
+++ b/src/main/java/org/bukkit/material/MaterialData.java
@@ -97,7 +97,7 @@ public class MaterialData implements Cloneable {
 
     @Override
     public boolean equals(@Nullable Object obj) {
-        if (obj != null && obj instanceof MaterialData) {
+        if (obj instanceof MaterialData) {
             MaterialData md = (MaterialData) obj;
 
             return (md.getItemType() == getItemType() && md.getData() == getData());
