From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 04:13:45 -0500
Subject: [PATCH] Remove redundant string builder


diff --git a/src/main/java/org/bukkit/configuration/MemorySection.java b/src/main/java/org/bukkit/configuration/MemorySection.java
index 2b3ee34a97e168342b5aee1c5108a04a0d7174f2..d24e9305139417f457cf2ab552fd9545cbbbd964 100644
--- a/src/main/java/org/bukkit/configuration/MemorySection.java
+++ b/src/main/java/org/bukkit/configuration/MemorySection.java
@@ -1033,13 +1033,11 @@ public class MemorySection implements ConfigurationSection {
     @Override
     public @NotNull String toString() {
         Configuration root = getRoot();
-        return new StringBuilder()
-            .append(getClass().getSimpleName())
-            .append("[path='")
-            .append(getCurrentPath())
-            .append("', root='")
-            .append(root == null ? null : root.getClass().getSimpleName())
-            .append("']")
-            .toString();
+        return getClass().getSimpleName() +
+            "[path='" +
+            getCurrentPath() +
+            "', root='" +
+            (root == null ? null : root.getClass().getSimpleName()) +
+            "']";
     }
 }
diff --git a/src/main/java/org/bukkit/help/IndexHelpTopic.java b/src/main/java/org/bukkit/help/IndexHelpTopic.java
index 143eba81a5c5a548c242062bebdc93c034a184dd..7c5d72fb554717049b44b470f77a40b0f805b50c 100644
--- a/src/main/java/org/bukkit/help/IndexHelpTopic.java
+++ b/src/main/java/org/bukkit/help/IndexHelpTopic.java
@@ -107,12 +107,11 @@ public class IndexHelpTopic extends HelpTopic {
      */
     @NotNull
     protected String buildIndexLine(@NotNull CommandSender sender, @NotNull HelpTopic topic) {
-        StringBuilder line = new StringBuilder();
-        line.append(ChatColor.GOLD);
-        line.append(topic.getName());
-        line.append(": ");
-        line.append(ChatColor.WHITE);
-        line.append(topic.getShortText());
-        return line.toString();
+	    final String line = ChatColor.GOLD +
+		    topic.getName() +
+		    ": " +
+		    ChatColor.WHITE +
+		    topic.getShortText();
+        return line;
     }
 }
diff --git a/src/main/java/org/bukkit/plugin/PluginLogger.java b/src/main/java/org/bukkit/plugin/PluginLogger.java
index e501c368ba0d3b17668acba01eec8656a3dd9196..1e0eed8dcb9cce410aad02472e5f1adfa70f6586 100644
--- a/src/main/java/org/bukkit/plugin/PluginLogger.java
+++ b/src/main/java/org/bukkit/plugin/PluginLogger.java
@@ -23,7 +23,7 @@ public class PluginLogger extends Logger {
     public PluginLogger(@NotNull Plugin context) {
         super(context.getClass().getCanonicalName(), null);
         String prefix = context.getDescription().getPrefix();
-        pluginName = prefix != null ? new StringBuilder().append("[").append(prefix).append("] ").toString() : "[" + context.getDescription().getName() + "] ";
+        pluginName = prefix != null ? "[" + prefix + "] " : "[" + context.getDescription().getName() + "] ";
         setParent(context.getServer().getLogger());
         setLevel(Level.ALL);
     }
diff --git a/src/main/java/org/bukkit/util/BoundingBox.java b/src/main/java/org/bukkit/util/BoundingBox.java
index 322b950db20067d90a14c5e1de686fb130e3b73c..5ee6e4b323085b8635b255ee7ca3ffc50c539de3 100644
--- a/src/main/java/org/bukkit/util/BoundingBox.java
+++ b/src/main/java/org/bukkit/util/BoundingBox.java
@@ -988,21 +988,20 @@ public class BoundingBox implements Cloneable, ConfigurationSerializable {
 
     @Override
     public @NotNull String toString() {
-        StringBuilder builder = new StringBuilder();
-        builder.append("BoundingBox [minX=");
-        builder.append(minX);
-        builder.append(", minY=");
-        builder.append(minY);
-        builder.append(", minZ=");
-        builder.append(minZ);
-        builder.append(", maxX=");
-        builder.append(maxX);
-        builder.append(", maxY=");
-        builder.append(maxY);
-        builder.append(", maxZ=");
-        builder.append(maxZ);
-        builder.append("]");
-        return builder.toString();
+        final String builder = "BoundingBox [minX=" +
+            minX +
+            ", minY=" +
+            minY +
+            ", minZ=" +
+            minZ +
+            ", maxX=" +
+            maxX +
+            ", maxY=" +
+            maxY +
+            ", maxZ=" +
+            maxZ +
+            "]";
+        return builder;
     }
 
     /**
diff --git a/src/main/java/org/bukkit/util/RayTraceResult.java b/src/main/java/org/bukkit/util/RayTraceResult.java
index 1f8df7ca62fedbdc82004c9abf28480735d98f2a..7feafd3885b20faae52cf55b007a7c555bbff120 100644
--- a/src/main/java/org/bukkit/util/RayTraceResult.java
+++ b/src/main/java/org/bukkit/util/RayTraceResult.java
@@ -147,16 +147,15 @@ public class RayTraceResult {
 
     @Override
     public @NotNull String toString() {
-        StringBuilder builder = new StringBuilder();
-        builder.append("RayTraceResult [hitPosition=");
-        builder.append(hitPosition);
-        builder.append(", hitBlock=");
-        builder.append(hitBlock);
-        builder.append(", hitBlockFace=");
-        builder.append(hitBlockFace);
-        builder.append(", hitEntity=");
-        builder.append(hitEntity);
-        builder.append("]");
-        return builder.toString();
+	    final String builder = "RayTraceResult [hitPosition=" +
+		    hitPosition +
+		    ", hitBlock=" +
+		    hitBlock +
+		    ", hitBlockFace=" +
+		    hitBlockFace +
+		    ", hitEntity=" +
+		    hitEntity +
+		    "]";
+        return builder;
     }
 }
