From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 04:41:47 -0500
Subject: [PATCH] Revert RegisteredServiceProvider from record class back to
 normal


diff --git a/src/main/java/org/bukkit/plugin/SimpleServicesManager.java b/src/main/java/org/bukkit/plugin/SimpleServicesManager.java
index 9a7117bc38b383103314a34b38741e00d6b532ab..2cc7ea23460575d573cba412c049e510338b5fed 100644
--- a/src/main/java/org/bukkit/plugin/SimpleServicesManager.java
+++ b/src/main/java/org/bukkit/plugin/SimpleServicesManager.java
@@ -81,7 +81,7 @@ public class SimpleServicesManager implements ServicesManager {
                         while (it2.hasNext()) {
                             RegisteredServiceProvider<?> registered = it2.next();
 
-                            if (registered.plugin().equals(plugin)) {
+                            if (registered.getPlugin().equals(plugin)) {
                                 it2.remove();
                                 unregisteredEvents.add(new ServiceUnregisterEvent(registered));
                             }
@@ -130,7 +130,7 @@ public class SimpleServicesManager implements ServicesManager {
                         while (it2.hasNext()) {
                             RegisteredServiceProvider<?> registered = it2.next();
 
-                            if (registered.provider() == provider) {
+                            if (registered.getProvider() == provider) {
                                 it2.remove();
                                 unregisteredEvents.add(new ServiceUnregisterEvent(registered));
                             }
@@ -172,7 +172,7 @@ public class SimpleServicesManager implements ServicesManager {
                         while (it2.hasNext()) {
                             RegisteredServiceProvider<?> registered = it2.next();
 
-                            if (registered.provider().equals(provider)) {
+                            if (registered.getProvider().equals(provider)) {
                                 it2.remove();
                                 unregisteredEvents.add(new ServiceUnregisterEvent(registered));
                             }
@@ -211,7 +211,7 @@ public class SimpleServicesManager implements ServicesManager {
             }
 
             // This should not be null!
-            return service.cast(registered.get(0).provider());
+            return service.cast(registered.get(0).getProvider());
         }
     }
 
@@ -252,7 +252,7 @@ public class SimpleServicesManager implements ServicesManager {
         synchronized (providers) {
             for (List<RegisteredServiceProvider<?>> registered : providers.values()) {
                 for (RegisteredServiceProvider<?> provider : registered) {
-                    if (provider.plugin().equals(plugin)) {
+                    if (provider.getPlugin().equals(plugin)) {
                         ret.add(provider);
                     }
                 }
