From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 03:04:59 -0500
Subject: [PATCH] Utilize sublist clear rather than looping


diff --git a/src/main/java/io/papermc/paper/util/PoiAccess.java b/src/main/java/io/papermc/paper/util/PoiAccess.java
index 07f40187f326a0d8d053be3070d1b39856546cbe..a2f0906b0eb93836978337a90012f041f636d18f 100644
--- a/src/main/java/io/papermc/paper/util/PoiAccess.java
+++ b/src/main/java/io/papermc/paper/util/PoiAccess.java
@@ -649,8 +649,8 @@ public final class PoiAccess {
         });
 
         // trim out any entries exceeding our maximum
-        for (int i = closestRecordsUnsorted.size() - 1; i >= max; --i) {
-            closestRecordsUnsorted.remove(i);
+        if (closestRecordsUnsorted.size() > max) {
+            closestRecordsUnsorted.subList(max, closestRecordsUnsorted.size()).clear();
         }
 
         // now we match perfectly what vanilla would have outputted, without having to search the whole radius (hopefully).
