From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 03:28:04 -0500
Subject: [PATCH] fixup boxing and unboxing


diff --git a/src/main/java/net/minecraft/nbt/ByteArrayTag.java b/src/main/java/net/minecraft/nbt/ByteArrayTag.java
index 5a02b5a903366c698a6493e5a31408548d73de6f..a333bb203b2b09a05668030ea1e1c23e44925765 100644
--- a/src/main/java/net/minecraft/nbt/ByteArrayTag.java
+++ b/src/main/java/net/minecraft/nbt/ByteArrayTag.java
@@ -67,7 +67,7 @@ public class ByteArrayTag extends CollectionTag<ByteTag> {
         for (int i = 0; i < list.size(); ++i) {
             Byte obyte = (Byte) list.get(i);
 
-            abyte[i] = (obyte == null ? Integer.valueOf(0) : obyte).byteValue();
+            abyte[i] = obyte == null ? 0 : obyte;
         }
 
         return abyte;
