From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 03:37:30 -0500
Subject: [PATCH] Utilize char where possible for indexOf


diff --git a/src/main/java/com/destroystokyo/paper/entity/ai/MobGoalHelper.java b/src/main/java/com/destroystokyo/paper/entity/ai/MobGoalHelper.java
index a66c25aefc8cc2b5dae120982f2ea52449228bf6..00daf2ea6924275a23f4d2c2ef8e22be27b5d05c 100644
--- a/src/main/java/com/destroystokyo/paper/entity/ai/MobGoalHelper.java
+++ b/src/main/java/com/destroystokyo/paper/entity/ai/MobGoalHelper.java
@@ -249,11 +249,11 @@ public class MobGoalHelper {
 
     public static String getUsableName(@NotNull Class<?> clazz) {
         String name = ObfHelper.INSTANCE.deobfClassName(clazz.getName());
-        name = name.substring(name.lastIndexOf(".") + 1);
+        name = name.substring(name.lastIndexOf('.') + 1);
         boolean flag = false;
         // inner classes
         if (name.contains("$")) {
-            String cut = name.substring(name.indexOf("$") + 1);
+            String cut = name.substring(name.indexOf('$') + 1);
             if (cut.length() <= 2) {
                 name = name.replace("Entity", "");
                 name = name.replace("$", "_");
diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index 9f38cc99658a3608af6977b1e985e2b3945e0911..2509fa9f8f5b224b796654c344ff93fbb4efc07d 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -1940,8 +1940,8 @@ public class ServerPlayer extends Player {
     public String getIpAddress() {
         String s = this.connection.connection.getRemoteAddress().toString();
 
-        s = s.substring(s.indexOf("/") + 1);
-        s = s.substring(0, s.indexOf(":"));
+        s = s.substring(s.indexOf('/') + 1);
+        s = s.substring(0, s.indexOf(':'));
         return s;
     }
 
