From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 03:39:20 -0500
Subject: [PATCH] Use isEmpty rather than equal check


diff --git a/src/main/java/net/minecraft/commands/arguments/selector/options/EntitySelectorOptions.java b/src/main/java/net/minecraft/commands/arguments/selector/options/EntitySelectorOptions.java
index 1deae10d44474007df65268601e9dbe49997b7cc..75f2590b7ff2dae853a062060b7c9153b9f9efb5 100644
--- a/src/main/java/net/minecraft/commands/arguments/selector/options/EntitySelectorOptions.java
+++ b/src/main/java/net/minecraft/commands/arguments/selector/options/EntitySelectorOptions.java
@@ -362,7 +362,7 @@ public class EntitySelectorOptions {
                 boolean bl = reader.shouldInvertValue();
                 String string = reader.getReader().readUnquotedString();
                 reader.addPredicate((entity) -> {
-                    if ("".equals(string)) {
+                    if (string != null && string.isEmpty()) {
                         return entity.getTags().isEmpty() != bl;
                     } else {
                         return entity.getTags().contains(string) != bl;
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index a03ceeb3dbd5ff25909275022163cc6e6a35ccd9..09d0c69101234e381af5382561e4d668dd5002a9 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1636,7 +1636,7 @@ public final class CraftServer implements Server {
             if (section != null) {
                 String name = section.getString("generator");
 
-                if ((name != null) && (!name.equals(""))) {
+                if ((name != null) && (!name.isEmpty())) {
                     String[] split = name.split(":", 2);
                     String id = (split.length > 1) ? split[1] : null;
                     Plugin plugin = this.pluginManager.getPlugin(split[0]);
@@ -1672,7 +1672,7 @@ public final class CraftServer implements Server {
             if (section != null) {
                 String name = section.getString("biome-provider");
 
-                if ((name != null) && (!name.equals(""))) {
+                if ((name != null) && (!name.isEmpty())) {
                     String[] split = name.split(":", 2);
                     String id = (split.length > 1) ? split[1] : null;
                     Plugin plugin = this.pluginManager.getPlugin(split[0]);
diff --git a/src/main/java/org/bukkit/craftbukkit/help/CustomHelpTopic.java b/src/main/java/org/bukkit/craftbukkit/help/CustomHelpTopic.java
index 2e1c94656421df17e4bac95c5e5ebcbaf715b419..18dbdaba6bb79bd3ec3d4b286e473fd0da0b0390 100644
--- a/src/main/java/org/bukkit/craftbukkit/help/CustomHelpTopic.java
+++ b/src/main/java/org/bukkit/craftbukkit/help/CustomHelpTopic.java
@@ -23,7 +23,7 @@ public class CustomHelpTopic extends HelpTopic {
             return true;
         }
 
-        if (!this.permissionNode.equals("")) {
+        if (!this.permissionNode.isEmpty()) {
             return sender.hasPermission(permissionNode);
         } else {
             return true;
diff --git a/src/main/java/org/bukkit/craftbukkit/help/SimpleHelpMap.java b/src/main/java/org/bukkit/craftbukkit/help/SimpleHelpMap.java
index 8b27744d27054044c10acf07501ef4f3631a9fe0..9494a708206ba12756e61e1850f2053bf855e4ca 100644
--- a/src/main/java/org/bukkit/craftbukkit/help/SimpleHelpMap.java
+++ b/src/main/java/org/bukkit/craftbukkit/help/SimpleHelpMap.java
@@ -58,7 +58,7 @@ public class SimpleHelpMap implements HelpMap {
 
     @Override
     public synchronized HelpTopic getHelpTopic(@NotNull String topicName) {
-        if (topicName.equals("")) {
+        if (topicName.isEmpty()) {
             return this.defaultTopic;
         }
 
