From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 03:36:15 -0500
Subject: [PATCH] Utilize enumset


diff --git a/src/main/java/net/minecraft/world/entity/LivingEntity.java b/src/main/java/net/minecraft/world/entity/LivingEntity.java
index 406e43351b377e8aff624ccd0d64781c271fde7b..41e764ca3574bc473347f6ec625925c97a9ad40f 100644
--- a/src/main/java/net/minecraft/world/entity/LivingEntity.java
+++ b/src/main/java/net/minecraft/world/entity/LivingEntity.java
@@ -10,13 +10,8 @@ import com.mojang.datafixers.util.Pair;
 import com.mojang.logging.LogUtils;
 import com.mojang.serialization.DataResult;
 import com.mojang.serialization.Dynamic;
-import java.util.Collection;
-import java.util.ConcurrentModificationException;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
-import java.util.UUID;
+
+import java.util.*;
 import java.util.function.Predicate;
 import javax.annotation.Nullable;
 import net.minecraft.BlockUtil;
@@ -124,9 +119,6 @@ import org.jetbrains.annotations.NotNull;
 import org.slf4j.Logger;
 
 // CraftBukkit start
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.Set;
 import com.google.common.base.Function;
 import org.bukkit.Location;
 import org.bukkit.craftbukkit.attribute.CraftAttributeMap;
@@ -1722,7 +1714,7 @@ public abstract class LivingEntity extends Entity {
 
     // Paper start
     protected boolean clearEquipmentSlots = true;
-    protected Set<EquipmentSlot> clearedEquipmentSlots = new java.util.HashSet<>();
+    protected Set<EquipmentSlot> clearedEquipmentSlots = EnumSet.noneOf(EquipmentSlot.class);
     protected org.bukkit.event.entity.EntityDeathEvent dropAllDeathLoot(DamageSource source) {
     // Paper end
         Entity entity = source.getEntity();
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftArmorStand.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftArmorStand.java
index ffb767eb6f3fec91349b7019a5ddd2b54f082be0..4dd0515a380e946dfba45079df17116d86cbcb1e 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftArmorStand.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftArmorStand.java
@@ -11,6 +11,8 @@ import org.bukkit.inventory.ItemStack;
 import org.bukkit.util.EulerAngle;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.EnumSet;
+
 
 public class CraftArmorStand extends CraftLivingEntity implements ArmorStand {
 
@@ -275,7 +277,7 @@ public class CraftArmorStand extends CraftLivingEntity implements ArmorStand {
 
     @Override
     public java.util.@NotNull Set<org.bukkit.inventory.EquipmentSlot> getDisabledSlots() {
-        java.util.Set<org.bukkit.inventory.EquipmentSlot> disabled = new java.util.HashSet<>();
+        java.util.Set<org.bukkit.inventory.EquipmentSlot> disabled = EnumSet.noneOf(EquipmentSlot.class);
         for (org.bukkit.inventory.EquipmentSlot slot : org.bukkit.inventory.EquipmentSlot.values()) {
             if (this.isSlotDisabled(slot)) {
                 disabled.add(slot);
