From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 04:04:10 -0500
Subject: [PATCH] Remove redundant checks


diff --git a/src/main/java/net/minecraft/CrashReport.java b/src/main/java/net/minecraft/CrashReport.java
index abe37c7c3c6f5ab73afd738ec78f06d7e4d2ed96..21604fb28c413dff0927afbeee231743ffffc672 100644
--- a/src/main/java/net/minecraft/CrashReport.java
+++ b/src/main/java/net/minecraft/CrashReport.java
@@ -58,7 +58,7 @@ public class CrashReport {
             this.uncategorizedStackTrace = (StackTraceElement[]) ArrayUtils.subarray(((CrashReportCategory) this.details.get(0)).getStacktrace(), 0, 1);
         }
 
-        if (this.uncategorizedStackTrace != null && this.uncategorizedStackTrace.length > 0) {
+        if (this.uncategorizedStackTrace.length > 0) {
             crashReportBuilder.append("-- Head --\n");
             crashReportBuilder.append("Thread: ").append(Thread.currentThread().getName()).append("\n");
             crashReportBuilder.append("Stacktrace:\n");
diff --git a/src/main/java/net/minecraft/world/entity/ExperienceOrb.java b/src/main/java/net/minecraft/world/entity/ExperienceOrb.java
index cf5c7e8557b0084039a94ef881a36aa9e3f58daf..03ea8a216ac5b5510344d33e9492c614ca252ec6 100644
--- a/src/main/java/net/minecraft/world/entity/ExperienceOrb.java
+++ b/src/main/java/net/minecraft/world/entity/ExperienceOrb.java
@@ -156,7 +156,7 @@ public class ExperienceOrb extends Entity {
             }
         }
 
-        if (this.followingPlayer != null && !cancelled) {
+        if (!cancelled) {
             // CraftBukkit end
             Vec3 vec3d = new Vec3(this.followingPlayer.getX() - this.getX(), this.followingPlayer.getY() + (double) this.followingPlayer.getEyeHeight() / 2.0D - this.getY(), this.followingPlayer.getZ() - this.getZ());
             double d0 = vec3d.lengthSqr();
diff --git a/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java b/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
index e5177109b5384e0fa4c5f163bb19c3e6e492b209..70d4d237ebb22c55a08573477c4acd7e7c65c4e4 100644
--- a/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
+++ b/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
@@ -510,7 +510,7 @@ public class WitherBoss extends Monster implements PowerableMob, RangedAttackMob
                 }
 
                 entity = source.getEntity();
-                if (entity != null && !(entity instanceof Player) && entity instanceof LivingEntity && ((LivingEntity) entity).getMobType() == this.getMobType()) {
+                if (!(entity instanceof Player) && entity instanceof LivingEntity && ((LivingEntity) entity).getMobType() == this.getMobType()) {
                     return false;
                 } else {
                     if (this.destroyBlocksTick <= 0) {
diff --git a/src/main/java/net/minecraft/world/entity/monster/Shulker.java b/src/main/java/net/minecraft/world/entity/monster/Shulker.java
index e4b687ed0b451b7fbbb215501ae268b01c56724f..2d9f9bdb0affc13aab33494c11a44d40019eda5d 100644
--- a/src/main/java/net/minecraft/world/entity/monster/Shulker.java
+++ b/src/main/java/net/minecraft/world/entity/monster/Shulker.java
@@ -604,7 +604,7 @@ public class Shulker extends AbstractGolem implements VariantHolder<Optional<Dye
     public DyeColor getColor() {
         byte b0 = ((Byte) this.entityData.get(Shulker.DATA_COLOR_ID)).byteValue();
 
-        return b0 != 16 && b0 <= 15 ? DyeColor.byId(b0) : null;
+        return b0 <= 15 ? DyeColor.byId(b0) : null;
     }
 
     private class ShulkerLookControl extends LookControl {
diff --git a/src/main/java/net/minecraft/world/entity/monster/Silverfish.java b/src/main/java/net/minecraft/world/entity/monster/Silverfish.java
index 37d3b670a93c26b87ffc5c93ad7828f45b4b0217..8501c50059631c6b8dc3c1592b9f8d11398ef4ef 100644
--- a/src/main/java/net/minecraft/world/entity/monster/Silverfish.java
+++ b/src/main/java/net/minecraft/world/entity/monster/Silverfish.java
@@ -127,7 +127,7 @@ public class Silverfish extends Monster {
         if (checkAnyLightMonsterSpawnRules(type, world, spawnReason, pos, random)) {
             Player entityhuman = world.getNearestPlayer((double) pos.getX() + 0.5D, (double) pos.getY() + 0.5D, (double) pos.getZ() + 0.5D, 5.0D, true);
 
-            return !(entityhuman != null && !entityhuman.affectsSpawning) && entityhuman == null; // Paper - Affects Spawning API
+            return entityhuman == null; // Paper - Affects Spawning API
         } else {
             return false;
         }
diff --git a/src/main/java/net/minecraft/world/entity/projectile/AbstractArrow.java b/src/main/java/net/minecraft/world/entity/projectile/AbstractArrow.java
index 6d05fa20d9d3415de4d3ce9d4fd46dc24ea24a4f..e82ce610b0af421b891679b41c1926e6e72df43a 100644
--- a/src/main/java/net/minecraft/world/entity/projectile/AbstractArrow.java
+++ b/src/main/java/net/minecraft/world/entity/projectile/AbstractArrow.java
@@ -426,7 +426,7 @@ public abstract class AbstractArrow extends Projectile {
                 }
 
                 this.doPostHurtEffects(entityliving);
-                if (entity1 != null && entityliving != entity1 && entityliving instanceof Player && entity1 instanceof ServerPlayer && !this.isSilent()) {
+                if (entityliving != entity1 && entityliving instanceof Player && entity1 instanceof ServerPlayer && !this.isSilent()) {
                     ((ServerPlayer) entity1).connection.send(new ClientboundGameEventPacket(ClientboundGameEventPacket.ARROW_HIT_PLAYER, 0.0F));
                 }
 
diff --git a/src/main/java/net/minecraft/world/level/block/entity/ConduitBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
index 05eab04e4aec4151018f25b59f92ddbbb4c09f87..68703fdfd17f3295dd37ea276f1cfcf9afee06e3 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
@@ -169,7 +169,7 @@ public class ConduitBlockEntity extends BlockEntity {
                     int i1 = Math.abs(j);
                     int j1 = Math.abs(k);
 
-                    if ((l > 1 || i1 > 1 || j1 > 1) && (i == 0 && (i1 == 2 || j1 == 2) || j == 0 && (l == 2 || j1 == 2) || k == 0 && (l == 2 || i1 == 2))) {
+                    if ((i == 0 && (i1 == 2 || j1 == 2) || j == 0 && (l == 2 || j1 == 2) || k == 0 && (l == 2 || i1 == 2))) {
                         BlockPos blockposition2 = pos.offset(i, j, k);
                         BlockState iblockdata = world.getBlockState(blockposition2);
                         Block[] ablock = ConduitBlockEntity.VALID_BLOCKS;
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
index 9125a1ab3ea57d29390d710cdfc8787a2883cb9f..d82748f7aaec88cd80d41de6b57aa5e781c8ce34 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
@@ -171,7 +171,7 @@ public class CraftOfflinePlayer implements OfflinePlayer, ConfigurationSerializa
 
     @Override
     public boolean equals(@Nullable Object obj) {
-        if (obj == null || !(obj instanceof OfflinePlayer)) {
+        if (!(obj instanceof OfflinePlayer)) {
             return false;
         }
 
