From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 03:59:09 -0500
Subject: [PATCH] Make inner classes static


diff --git a/src/main/java/net/minecraft/world/level/chunk/LevelChunk.java b/src/main/java/net/minecraft/world/level/chunk/LevelChunk.java
index bb86b48462ee564449776c1226cad25d982579fd..32eb133a4e61d1aede5f7aef799c9e72ceff9eb8 100644
--- a/src/main/java/net/minecraft/world/level/chunk/LevelChunk.java
+++ b/src/main/java/net/minecraft/world/level/chunk/LevelChunk.java
@@ -1187,7 +1187,7 @@ public class LevelChunk extends ChunkAccess {
                     chunk_d.rebind(tickingblockentity);
                     return chunk_d;
                 } else if (this.isInLevel()) {
-                    LevelChunk.RebindableTickingBlockEntityWrapper chunk_d1 = new LevelChunk.RebindableTickingBlockEntityWrapper(tickingblockentity);
+                    LevelChunk.RebindableTickingBlockEntityWrapper chunk_d1 = new RebindableTickingBlockEntityWrapper(tickingblockentity);
 
                     this.level.addBlockEntityTicker(chunk_d1);
                     return chunk_d1;
@@ -1224,7 +1224,7 @@ public class LevelChunk extends ChunkAccess {
         private EntityCreationType() {}
     }
 
-    private class RebindableTickingBlockEntityWrapper implements TickingBlockEntity {
+    private static class RebindableTickingBlockEntityWrapper implements TickingBlockEntity {
 
         private TickingBlockEntity ticker;
 
diff --git a/src/main/java/org/bukkit/craftbukkit/help/SimpleHelpMap.java b/src/main/java/org/bukkit/craftbukkit/help/SimpleHelpMap.java
index 573f6094c34111bc6ceb794a6d27a6a3a427f00f..c24cb468e21d4ceb9294a637fa1f626884fa5cb7 100644
--- a/src/main/java/org/bukkit/craftbukkit/help/SimpleHelpMap.java
+++ b/src/main/java/org/bukkit/craftbukkit/help/SimpleHelpMap.java
@@ -232,7 +232,7 @@ public class SimpleHelpMap implements HelpMap {
         this.topicFactoryMap.put(commandClass, factory);
     }
 
-    private class IsCommandTopicPredicate implements Predicate<HelpTopic> {
+    private static class IsCommandTopicPredicate implements Predicate<HelpTopic> {
 
         @Override
         public boolean apply(@NotNull HelpTopic topic) {
