From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 04:03:45 -0500
Subject: [PATCH] Simplify comparator


diff --git a/src/main/java/net/minecraft/server/PlayerAdvancements.java b/src/main/java/net/minecraft/server/PlayerAdvancements.java
index fcbb8c9f87d77dc362b8d67e306a90a394142b26..46fff641abef20753c4f72a3d5fb71769eeb964a 100644
--- a/src/main/java/net/minecraft/server/PlayerAdvancements.java
+++ b/src/main/java/net/minecraft/server/PlayerAdvancements.java
@@ -173,7 +173,7 @@ public class PlayerAdvancements {
                         throw new JsonParseException("Found null for advancements");
                     }
 
-                    Stream<Entry<ResourceLocation, AdvancementProgress>> stream = map.entrySet().stream().sorted(Comparator.comparing(Entry::getValue));
+                    Stream<Entry<ResourceLocation, AdvancementProgress>> stream = map.entrySet().stream().sorted(Entry.comparingByValue());
                     Iterator iterator = ((List) stream.collect(Collectors.toList())).iterator();
 
                     while (iterator.hasNext()) {
