From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 04:07:49 -0500
Subject: [PATCH] Unfold ifPresent


diff --git a/src/main/java/net/minecraft/world/effect/MobEffectInstance.java b/src/main/java/net/minecraft/world/effect/MobEffectInstance.java
index 1cdbd27c3d36993f205e05480a82b48da4b0449e..55d4462f890018c544a1eb6642abfa15d1421494 100644
--- a/src/main/java/net/minecraft/world/effect/MobEffectInstance.java
+++ b/src/main/java/net/minecraft/world/effect/MobEffectInstance.java
@@ -249,10 +249,8 @@ public class MobEffectInstance implements Comparable<MobEffectInstance> {
             nbt.put("HiddenEffect", compoundTag);
         }
 
-        this.factorData.ifPresent((factorCalculationData) -> {
-            MobEffectInstance.FactorData.CODEC.encodeStart(NbtOps.INSTANCE, factorCalculationData).resultOrPartial(LOGGER::error).ifPresent((factorCalculationDataNbt) -> {
-                nbt.put("FactorCalculationData", factorCalculationDataNbt);
-            });
+        this.factorData.flatMap(factorCalculationData -> FactorData.CODEC.encodeStart(NbtOps.INSTANCE, factorCalculationData).resultOrPartial(LOGGER::error)).ifPresent((factorCalculationDataNbt) -> {
+	        nbt.put("FactorCalculationData", factorCalculationDataNbt);
         });
     }
 
