From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 04:09:22 -0500
Subject: [PATCH] Remove redundant string operations


diff --git a/src/main/java/org/bukkit/craftbukkit/legacy/CraftLegacy.java b/src/main/java/org/bukkit/craftbukkit/legacy/CraftLegacy.java
index df96f7f55d2e6654f0bcbe216e68dde0b534f5b1..d19a28226b873f327442fad5069119db2526113e 100644
--- a/src/main/java/org/bukkit/craftbukkit/legacy/CraftLegacy.java
+++ b/src/main/java/org/bukkit/craftbukkit/legacy/CraftLegacy.java
@@ -438,6 +438,6 @@ public final class CraftLegacy {
     }
 
     public static void main(String[] args) {
-        System.err.println("");
+        System.err.println();
     }
 }
