From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 04:06:00 -0500
Subject: [PATCH] Use Math.max


diff --git a/src/main/java/alternate/current/wire/WireHandler.java b/src/main/java/alternate/current/wire/WireHandler.java
index 71526836cf17aa7e2c144380e458299be61436b5..f5908780bfc5fb3548b50226505f7e7a4f9179f2 100644
--- a/src/main/java/alternate/current/wire/WireHandler.java
+++ b/src/main/java/alternate/current/wire/WireHandler.java
@@ -822,15 +822,9 @@ public class WireHandler {
      * Queue the given wire for the breadth-first search as a root.
      */
     private void searchRoot(@NotNull WireNode wire) {
-        int iBackupFlowDir;
+        int iBackupFlowDir = Math.max(wire.connections.iFlowDir, 0);
 
-        if (wire.connections.iFlowDir < 0) {
-            iBackupFlowDir = 0;
-        } else {
-            iBackupFlowDir = wire.connections.iFlowDir;
-        }
-
-        search(wire, true, iBackupFlowDir);
+	    search(wire, true, iBackupFlowDir);
     }
 
     /**
diff --git a/src/main/java/ca/spottedleaf/dataconverter/minecraft/MCDataConverter.java b/src/main/java/ca/spottedleaf/dataconverter/minecraft/MCDataConverter.java
index 5a8ebb403e65afd3c7eafd5ce9944978a60231d0..8bf3948b3871c8dd0c69d607c4f858feeb819f08 100644
--- a/src/main/java/ca/spottedleaf/dataconverter/minecraft/MCDataConverter.java
+++ b/src/main/java/ca/spottedleaf/dataconverter/minecraft/MCDataConverter.java
@@ -44,7 +44,7 @@ public final class MCDataConverter {
     public static <T, R> R convert(final @NotNull DataType<T, R> type, final T data, int fromVersion, final int toVersion) {
         Object ret = data;
 
-        long currentVersion = DataConverter.encodeVersions(fromVersion < 99 ? 99 : fromVersion, Integer.MAX_VALUE);
+        long currentVersion = DataConverter.encodeVersions(Math.max(fromVersion, 99), Integer.MAX_VALUE);
         final long nextVersion = DataConverter.encodeVersions(toVersion, Integer.MAX_VALUE);
 
         for (int i = 0, len = BREAKPOINTS.size(); i < len; ++i) {
diff --git a/src/main/java/com/destroystokyo/paper/util/RedstoneWireTurbo.java b/src/main/java/com/destroystokyo/paper/util/RedstoneWireTurbo.java
index 04a43c85a62f0e968d6ce980eba1b39c2cb1e395..61c48393525c84f95a5465cf132c59a66bde7402 100644
--- a/src/main/java/com/destroystokyo/paper/util/RedstoneWireTurbo.java
+++ b/src/main/java/com/destroystokyo/paper/util/RedstoneWireTurbo.java
@@ -955,6 +955,6 @@ public class RedstoneWireTurbo {
     private static int getMaxCurrentStrength(final @NotNull UpdateNode upd, final int strength) {
         if (upd.type != UpdateNode.Type.REDSTONE) return strength;
         final int i = upd.currentState.getValue(RedStoneWireBlock.POWER).intValue();
-        return i > strength ? i : strength;
+        return Math.max(i, strength);
     }
 }
diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java b/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
index f8dcb18f460400bd82c4be8a4eea9c4b18c009ad..54131cb655ac35291ccd66d3baca3eeb41e3196e 100644
--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
@@ -454,7 +454,7 @@ public class CraftBlock implements Block {
         } else {
             int j = iblockdata.<Integer>getValue(RedStoneWireBlock.POWER).intValue();
 
-            return j > i ? j : i;
+            return Math.max(j, i);
         }
     }
 
