From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 04:10:30 -0500
Subject: [PATCH] Remove redundant filter


diff --git a/src/main/java/net/minecraft/world/entity/ai/village/poi/PoiManager.java b/src/main/java/net/minecraft/world/entity/ai/village/poi/PoiManager.java
index 204bdfa22b2ef770379bb50c41a347bed823f665..82f68bebc227a3dbd846fed237c77a6710aa0b63 100644
--- a/src/main/java/net/minecraft/world/entity/ai/village/poi/PoiManager.java
+++ b/src/main/java/net/minecraft/world/entity/ai/village/poi/PoiManager.java
@@ -362,14 +362,13 @@ public class PoiManager extends SectionStorage<PoiSection> {
     }
 
     public void ensureLoadedAndValid(LevelReader world, BlockPos pos, int radius) {
+        // Paper - rewrite chunk system
         SectionPos.aroundChunk(new ChunkPos(pos), Math.floorDiv(radius, 16), this.levelHeightAccessor.getMinSection(), this.levelHeightAccessor.getMaxSection()).map((sectionPos) -> {
             return Pair.of(sectionPos, this.getOrLoad(sectionPos.asLong()));
         }).filter((pair) -> {
             return !pair.getSecond().map(PoiSection::isValid).orElse(Boolean.FALSE).booleanValue();
         }).map((pair) -> {
             return pair.getFirst().chunk();
-        }).filter((chunkPos) -> {
-            return true; // Paper - rewrite chunk system
         }).forEach((chunkPos) -> {
             world.getChunk(chunkPos.x, chunkPos.z, ChunkStatus.EMPTY);
         });
