From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 04:14:48 -0500
Subject: [PATCH] Remove redundant semicolon


diff --git a/build.gradle.kts b/build.gradle.kts
index 360e6f89dd2ab81dae41bf33efd8e9681a50f4c1..b85d59a1a00af1ad0d0b66ce3202ab5aa7ed3989 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -58,7 +58,7 @@ dependencies {
     testImplementation("org.junit.jupiter:junit-jupiter-params:5.9.0")
     testImplementation("org.hamcrest:hamcrest-library:2.2")
 
-    implementation("io.netty:netty-all:4.1.84.Final"); // Paper - Bump netty
+    implementation("io.netty:netty-all:4.1.84.Final") // Paper - Bump netty
 }
 
 val craftbukkitPackageVersion = "1_19_R1" // Paper
