From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 04:29:51 -0500
Subject: [PATCH] Fixup patches


diff --git a/src/main/java/net/minecraft/world/entity/ai/sensing/TemptingSensor.java b/src/main/java/net/minecraft/world/entity/ai/sensing/TemptingSensor.java
index 1ec44433a59a1c3bbdb66bbeffc8737320476270..16f1c5bd485dfc8e8800692ae6bf1b28ca5eb923 100644
--- a/src/main/java/net/minecraft/world/entity/ai/sensing/TemptingSensor.java
+++ b/src/main/java/net/minecraft/world/entity/ai/sensing/TemptingSensor.java
@@ -5,10 +5,10 @@ import java.util.Comparator;
 import java.util.List;
 import java.util.Objects;
 import java.util.Set;
-import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.entity.Entity;
+import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.world.entity.EntitySelector;
 import net.minecraft.world.entity.PathfinderMob;
 import net.minecraft.world.entity.ai.Brain;
@@ -46,7 +46,7 @@ public class TemptingSensor extends Sensor<PathfinderMob> {
         });
 
         Objects.requireNonNull(entity);
-        List<Player> list = stream.sorted(Comparator.comparingDouble(entity::distanceToSqr)).toList();
+        List<ServerPlayer> list = stream.sorted(Comparator.comparingDouble(entity::distanceToSqr)).toList();
 
         if (!list.isEmpty()) {
             Player entityhuman = list.get(0);
diff --git a/src/main/java/net/minecraft/world/level/block/DispenserBlock.java b/src/main/java/net/minecraft/world/level/block/DispenserBlock.java
index 78dec62e80c4e5effcd1b9c09d37e8a12a486b64..0077dd2a87a18854a053cdd322d33e0240c0c3a1 100644
--- a/src/main/java/net/minecraft/world/level/block/DispenserBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/DispenserBlock.java
@@ -154,7 +154,7 @@ public class DispenserBlock extends BaseEntityBlock {
             BlockEntity tileentity = world.getBlockEntity(pos);
 
             if (tileentity instanceof DispenserBlockEntity) {
-                Containers.dropContents(world, pos, tileentity);
+                Containers.dropContents(world, pos, (Container) tileentity);
                 world.updateNeighbourForOutputSignal(pos, this);
             }
 
