From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 19:48:37 -0500
Subject: [PATCH] Add mockito


diff --git a/build.gradle.kts b/build.gradle.kts
index 74d9c8cdc45f789a0c530d38a0057f0f17902a10..85a2b3d8af891c58900f71507f083101616ceadd 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -61,6 +61,7 @@ dependencies {
     testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.9.0")
     testImplementation("org.junit.jupiter:junit-jupiter-params:5.9.0")
     testImplementation("org.hamcrest:hamcrest-library:2.2")
+    testImplementation("org.mockito:mockito-core:4.8.0")
 
     implementation("io.netty:netty-all:4.1.84.Final") // Paper - Bump netty
 }
