From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 22 Oct 2022 04:14:17 -0500
Subject: [PATCH] Remove redundant escape characters


diff --git a/src/main/java/net/minecraft/server/Bootstrap.java b/src/main/java/net/minecraft/server/Bootstrap.java
index 0af4a1dff11e6d4610bf74914efcf01eea170dca..60a10d8d6a56ac337e63f1148e294b85d196dcf9 100644
--- a/src/main/java/net/minecraft/server/Bootstrap.java
+++ b/src/main/java/net/minecraft/server/Bootstrap.java
@@ -98,7 +98,7 @@ public class Bootstrap {
                 BlockStateData.register(1023, "{Name:'minecraft:oak_sign',Properties:{rotation:'15'}}", "{Name:'minecraft:standing_sign',Properties:{rotation:'15'}}");
                 ItemIdFix.ITEM_NAMES.put(323, "minecraft:oak_sign");
 
-                BlockStateData.register(1440, "{Name:\'minecraft:portal\',Properties:{axis:\'x\'}}", "{Name:\'minecraft:portal\',Properties:{axis:\'x\'}}");
+                BlockStateData.register(1440, "{Name:'minecraft:portal',Properties:{axis:'x'}}", "{Name:\'minecraft:portal\',Properties:{axis:\'x\'}}");
 
                 ItemIdFix.ITEM_NAMES.put(409, "minecraft:prismarine_shard");
                 ItemIdFix.ITEM_NAMES.put(410, "minecraft:prismarine_crystals");
diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 1549a8384426630743b422808afa45c258fa594a..d6f0fcf433ccae9505947dad84bc5c052864e501 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -3653,7 +3653,7 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
                     this.getCraftPlayer().addChannel(channel);
                 }
             } catch (Exception ex) {
-                ServerGamePacketListenerImpl.LOGGER.error("Couldn\'t register custom payload", ex);
+                ServerGamePacketListenerImpl.LOGGER.error("Couldn't register custom payload", ex);
                 this.disconnect("Invalid payload REGISTER!", org.bukkit.event.player.PlayerKickEvent.Cause.INVALID_PAYLOAD); // Paper - kick event cause
             }
         } else if (packet.identifier.equals(CUSTOM_UNREGISTER)) {
@@ -3663,7 +3663,7 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
                     this.getCraftPlayer().removeChannel(channel);
                 }
             } catch (Exception ex) {
-                ServerGamePacketListenerImpl.LOGGER.error("Couldn\'t unregister custom payload", ex);
+                ServerGamePacketListenerImpl.LOGGER.error("Couldn't unregister custom payload", ex);
                 this.disconnect("Invalid payload UNREGISTER!", org.bukkit.event.player.PlayerKickEvent.Cause.INVALID_PAYLOAD); // Paper - kick event cause
             }
         } else {
@@ -3681,7 +3681,7 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
                 // Paper end
                 this.cserver.getMessenger().dispatchIncomingMessage(this.player.getBukkitEntity(), packet.identifier.toString(), data);
             } catch (Exception ex) {
-                ServerGamePacketListenerImpl.LOGGER.error("Couldn\'t dispatch custom payload", ex);
+                ServerGamePacketListenerImpl.LOGGER.error("Couldn't dispatch custom payload", ex);
                 this.disconnect("Invalid custom payload!", org.bukkit.event.player.PlayerKickEvent.Cause.INVALID_PAYLOAD); // Paper - kick event cause
             }
         }
diff --git a/src/main/java/org/spigotmc/SpigotConfig.java b/src/main/java/org/spigotmc/SpigotConfig.java
index a067a5cc30bcd2dbecd914b75504fa105b1517dc..18ebebc76d630f531706b8036b4db5d1ef361f4b 100644
--- a/src/main/java/org/spigotmc/SpigotConfig.java
+++ b/src/main/java/org/spigotmc/SpigotConfig.java
@@ -201,7 +201,7 @@ public class SpigotConfig
     public static String unknownCommandMessage;
     public static String serverFullMessage;
     public static @NotNull String outdatedClientMessage = "Outdated client! Please use {0}";
-    public static @NotNull String outdatedServerMessage = "Outdated server! I\'m still on {0}";
+    public static @NotNull String outdatedServerMessage = "Outdated server! I'm still on {0}";
     private static @NotNull String transform(@NotNull String s)
     {
         return ChatColor.translateAlternateColorCodes( '&', s ).replaceAll( "\\\\n", "\n" );
