From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sun, 13 Nov 2022 07:36:11 -0600
Subject: [PATCH] Fix tests


diff --git a/src/main/java/org/bukkit/craftbukkit/advancement/CraftAdvancement.java b/src/main/java/org/bukkit/craftbukkit/advancement/CraftAdvancement.java
index b2a159f6716228ac5907edcdf2acab018a406914..6039955122e1cda261a3ceb18371e7f30bb32231 100644
--- a/src/main/java/org/bukkit/craftbukkit/advancement/CraftAdvancement.java
+++ b/src/main/java/org/bukkit/craftbukkit/advancement/CraftAdvancement.java
@@ -4,6 +4,7 @@ import java.util.Collection;
 import java.util.Collections;
 import net.minecraft.advancements.Advancement;
 import org.bukkit.NamespacedKey;
+import org.bukkit.advancement.AdvancementDisplay;
 import org.bukkit.craftbukkit.util.CraftNamespacedKey;
 import org.jetbrains.annotations.NotNull;
 
