From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sun, 13 Nov 2022 17:45:51 -0600
Subject: [PATCH] Add kotlin reflect


diff --git a/build.gradle.kts b/build.gradle.kts
index 57080385a5431ea1474ef228dd5ca7ceb2420be0..0f3f569543a3fba6f539ca04845047a218927878 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -21,6 +21,7 @@ dependencies {
 
     implementation(kotlin("bom"))
     implementation(kotlin("stdlib-jdk8"))
+    implementation(kotlin("reflect"))
     // CatPaper end
 
     // Paper start
