From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sun, 13 Nov 2022 17:45:51 -0600
Subject: [PATCH] Add kotlin reflect


diff --git a/build.gradle.kts b/build.gradle.kts
index e97d95135b7581fd015d8c3f5f5f5d155d5af970..7c45a0638ac02dbe0c51388001aed837228916e2 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -21,6 +21,7 @@ dependencies {
 
     implementation(kotlin("bom"))
     implementation(kotlin("stdlib-jdk8"))
+    implementation(kotlin("reflect"))
     // CatPaper end
 
     // Paper start
