From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Fri, 25 Nov 2022 01:34:02 -0600
Subject: [PATCH] Fixes


diff --git a/src/main/java/net/minecraft/world/level/pathfinder/FlyNodeEvaluator.java b/src/main/java/net/minecraft/world/level/pathfinder/FlyNodeEvaluator.java
index fc2c81385cdfb4c4a910becbd6967ee481cea059..a1dde1c741ce6099f267c5906c2227ebcd2bf62d 100644
--- a/src/main/java/net/minecraft/world/level/pathfinder/FlyNodeEvaluator.java
+++ b/src/main/java/net/minecraft/world/level/pathfinder/FlyNodeEvaluator.java
@@ -35,11 +35,11 @@ public class FlyNodeEvaluator extends WalkNodeEvaluator {
         int i;
         if (this.canFloat() && this.mob.isInWater()) {
             i = this.mob.getBlockY();
-            BlockPos.MutableBlockPos mutableBlockPos = new BlockPos.MutableBlockPos(this.mob.getX(), (double)i, this.mob.getZ());
+            BlockPos.MutableBlockPos mutableBlockPos = new BlockPos.MutableBlockPos(this.mob.getX(), i, this.mob.getZ());
 
             for(BlockState blockState = this.level.getBlockState(mutableBlockPos); blockState.is(Blocks.WATER); blockState = this.level.getBlockState(mutableBlockPos)) {
                 ++i;
-                mutableBlockPos.set(this.mob.getX(), (double)i, this.mob.getZ());
+                mutableBlockPos.set(this.mob.getX(), i, this.mob.getZ());
             }
         } else {
             i = Mth.floor(this.mob.getY() + 0.5D);
