From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Fri, 25 Nov 2022 01:36:19 -0600
Subject: [PATCH] Fix crashes


diff --git a/src/main/java/net/minecraft/world/level/pathfinder/NodeEvaluator.java b/src/main/java/net/minecraft/world/level/pathfinder/NodeEvaluator.java
index 96739b9795846e6b8883e5d2743f0e95ce198ee3..96fd48359509fa5ffc66f1ccd675b13eb961bdc7 100644
--- a/src/main/java/net/minecraft/world/level/pathfinder/NodeEvaluator.java
+++ b/src/main/java/net/minecraft/world/level/pathfinder/NodeEvaluator.java
@@ -11,7 +11,7 @@ import net.minecraft.world.level.PathNavigationRegion;
 
 public abstract class NodeEvaluator {
     protected PathNavigationRegion level;
-    protected Mob mob;
+    protected volatile Mob mob;
     protected final Int2ObjectMap<Node> nodes = new Int2ObjectOpenHashMap<>();
     protected int entityWidth;
     protected int entityHeight;
@@ -30,8 +30,8 @@ public abstract class NodeEvaluator {
     }
 
     public void done() {
-        this.level = null;
-        this.mob = null;
+        //this.level = null;
+        //this.mob = null;
     }
 
     @Nullable
