From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Mon, 14 Nov 2022 20:25:45 -0600
Subject: [PATCH] Fix world saving


diff --git a/src/main/java/net/minecraft/world/level/levelgen/LegacyRandomSource.java b/src/main/java/net/minecraft/world/level/levelgen/LegacyRandomSource.java
index aa5e4f70619fe7ee0f0e6bb58dee842c946bd123..29f827c6655d46c2db59b862df72def2739b77aa 100644
--- a/src/main/java/net/minecraft/world/level/levelgen/LegacyRandomSource.java
+++ b/src/main/java/net/minecraft/world/level/levelgen/LegacyRandomSource.java
@@ -33,7 +33,9 @@ public class LegacyRandomSource implements BitRandomSource {
         //if (!this.seed.compareAndSet(this.seed.get(), (seed ^ 25214903917L) & 281474976710655L)) {
             //throw ThreadingDetector.makeThreadingException("LegacyRandomSource", (Thread)null);
         //} else {
-            this.gaussianSource.reset();
+
+        this.seed.compareAndSet(this.seed.get(), (seed ^ 25214903917L) & 281474976710655L);
+        this.gaussianSource.reset();
         //}
     }
 
@@ -44,7 +46,8 @@ public class LegacyRandomSource implements BitRandomSource {
        // if (!this.seed.compareAndSet(l, m)) {
             //throw ThreadingDetector.makeThreadingException("LegacyRandomSource", (Thread)null);
         //} else {
-            return (int)(m >> 48 - bits);
+        this.seed.compareAndSet(l, m);
+        return (int)(m >> 48 - bits);
         //}
     }
 
