From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Sat, 17 Dec 2022 01:46:37 -0600
Subject: [PATCH] Add is testing


diff --git a/src/main/java/io/papermc/paper/command/PaperCommands.java b/src/main/java/io/papermc/paper/command/PaperCommands.java
index b1e1a5bf037476449dea7687b0215ae7cfe0a731..2bbf253fa2cb088fd091af315ccb71a0eb87320c 100644
--- a/src/main/java/io/papermc/paper/command/PaperCommands.java
+++ b/src/main/java/io/papermc/paper/command/PaperCommands.java
@@ -2,24 +2,30 @@ package io.papermc.paper.command;
 
 import net.minecraft.server.MinecraftServer;
 import org.bukkit.command.Command;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.framework.qual.DefaultQualifier;
 
 import java.util.HashMap;
 import java.util.Map;
-import org.checkerframework.checker.nullness.qual.NonNull;
-import org.checkerframework.framework.qual.DefaultQualifier;
+
 
 @DefaultQualifier(NonNull.class)
 public final class PaperCommands {
 
+    private static final boolean IS_TESTING = false;
+
     private PaperCommands() {
     }
 
     private static final Map<String, Command> COMMANDS = new HashMap<>();
+
     static {
         COMMANDS.put("paper", new PaperCommand("paper"));
         COMMANDS.put("mspt", new MSPTCommand("mspt"));
         COMMANDS.put("mem", new MemCommand("mem"));
-        COMMANDS.put("spawnAllMobs", new SpawnAllMobsCommand("spawnAllMobs"));
+        if (IS_TESTING) {
+            COMMANDS.put("spawnAllMobs", new SpawnAllMobsCommand("spawnAllMobs"));
+        }
     }
 
     public static void registerCommands(final MinecraftServer server) {
