From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Fri, 25 Nov 2022 11:21:09 -0600
Subject: [PATCH] Fix crashes


diff --git a/src/main/java/net/minecraft/world/damagesource/CombatTracker.java b/src/main/java/net/minecraft/world/damagesource/CombatTracker.java
index ba96589aab8be0a90144f73f7779769146c7a37e..f8586b3fe365404f1173a9eca57b503bbccc9cf3 100644
--- a/src/main/java/net/minecraft/world/damagesource/CombatTracker.java
+++ b/src/main/java/net/minecraft/world/damagesource/CombatTracker.java
@@ -3,6 +3,7 @@ package net.minecraft.world.damagesource;
 import com.google.common.collect.Lists;
 import java.util.List;
 import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
 import net.minecraft.network.chat.Component;
@@ -17,7 +18,7 @@ import net.minecraft.world.level.block.state.BlockState;
 public class CombatTracker {
     public static final int RESET_DAMAGE_STATUS_TIME = 100;
     public static final int RESET_COMBAT_STATUS_TIME = 300;
-    private final List<CombatEntry> entries = Lists.newArrayList();
+    private final List<CombatEntry> entries = new CopyOnWriteArrayList<>();
     private final LivingEntity mob;
     private int lastDamageTime;
     private int combatStartTime;
@@ -137,7 +138,7 @@ public class CombatTracker {
             }
         }
 
-        return (LivingEntity)(player != null && g >= f / 3.0F ? player : livingEntity);
+        return player != null && g >= f / 3.0F ? player : livingEntity;
     }
 
     @Nullable
