From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Mon, 14 Nov 2022 21:32:39 -0600
Subject: [PATCH] Fix experience orb crash


diff --git a/src/main/java/net/minecraft/world/entity/ExperienceOrb.java b/src/main/java/net/minecraft/world/entity/ExperienceOrb.java
index b29a355a421acb45bbf02dcd3ede2a43bef1b07b..a44df483996ec38fd6f4d35e6f04b18b783935a7 100644
--- a/src/main/java/net/minecraft/world/entity/ExperienceOrb.java
+++ b/src/main/java/net/minecraft/world/entity/ExperienceOrb.java
@@ -143,21 +143,25 @@ public class ExperienceOrb extends Entity {
 
         // CraftBukkit start
         boolean cancelled = false;
+
+        Player thisTickFollowingPlayer = prevTarget;
+
         if (this.followingPlayer != prevTarget) {
             EntityTargetLivingEntityEvent event = CraftEventFactory.callEntityTargetLivingEvent(this, followingPlayer, (this.followingPlayer != null) ? EntityTargetEvent.TargetReason.CLOSEST_PLAYER : EntityTargetEvent.TargetReason.FORGOT_TARGET);
             LivingEntity target = (event.getTarget() == null) ? null : ((org.bukkit.craftbukkit.entity.CraftLivingEntity) event.getTarget()).getHandle();
             cancelled = event.isCancelled();
 
-            if (cancelled) {
-                this.followingPlayer = prevTarget;
-            } else {
-                this.followingPlayer = (target instanceof Player) ? (Player) target : null;
+            if (!cancelled) {
+               thisTickFollowingPlayer = (target instanceof Player) ? (Player) target : null;
             }
+
+            this.followingPlayer = thisTickFollowingPlayer;
         }
 
-        if (!cancelled) {
+        if (!cancelled && thisTickFollowingPlayer != null) {
+
             // CraftBukkit end
-            Vec3 vec3d = new Vec3(this.followingPlayer.getX() - this.getX(), this.followingPlayer.getY() + (double) this.followingPlayer.getEyeHeight() / 2.0D - this.getY(), this.followingPlayer.getZ() - this.getZ());
+            Vec3 vec3d = new Vec3(thisTickFollowingPlayer.getX() - this.getX(), thisTickFollowingPlayer.getY() + (double) thisTickFollowingPlayer.getEyeHeight() / 2.0D - this.getY(), thisTickFollowingPlayer.getZ() - this.getZ());
             double d0 = vec3d.lengthSqr();
 
             if (d0 < 64.0D) {
