From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Fri, 25 Nov 2022 13:35:04 -0600
Subject: [PATCH] Fix crashes


diff --git a/src/main/java/net/minecraft/world/level/redstone/CollectingNeighborUpdater.java b/src/main/java/net/minecraft/world/level/redstone/CollectingNeighborUpdater.java
index 526abd6cd5b82e1b0fe8315bf866b3f9c85c3524..5d1b14f30690b192fa408fd76ab15077d0084c59 100644
--- a/src/main/java/net/minecraft/world/level/redstone/CollectingNeighborUpdater.java
+++ b/src/main/java/net/minecraft/world/level/redstone/CollectingNeighborUpdater.java
@@ -49,9 +49,10 @@ public class CollectingNeighborUpdater implements NeighborUpdater {
 
     private void runUpdates() {
 
-        int count = 0;
+        //int count = 0;
 
-        while (count < this.maxChainedNeighborUpdates) {
+        //while (count < this.maxChainedNeighborUpdates) {
+        while (true) {
 
             final CollectingNeighborUpdater.NeighborUpdates entry = this.stack.peek();
             if (entry == null) {
@@ -62,8 +63,9 @@ public class CollectingNeighborUpdater implements NeighborUpdater {
                 stack.remove(entry);
             }
 
-            count++;
+            //count++;
         }
+       // }
     }
 
     static record FullNeighborUpdate(BlockState state, BlockPos pos, Block block, BlockPos neighborPos, boolean movedByPiston) implements CollectingNeighborUpdater.NeighborUpdates {
