From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Mon, 14 Nov 2022 23:52:52 -0600
Subject: [PATCH] Fix polarbear


diff --git a/src/main/java/net/minecraft/world/entity/animal/PolarBear.java b/src/main/java/net/minecraft/world/entity/animal/PolarBear.java
index b1e05acfb0f0d6517682c12e1ba06a685f0280db..15fb3fe37463c64ccf7831e17fe385b4a2a7c643 100644
--- a/src/main/java/net/minecraft/world/entity/animal/PolarBear.java
+++ b/src/main/java/net/minecraft/world/entity/animal/PolarBear.java
@@ -216,7 +216,7 @@ public class PolarBear extends Animal implements NeutralMob {
 
     @Override
     public boolean doHurtTarget(Entity target) {
-        boolean bl = target.hurt(DamageSource.mobAttack(this), (float)((int)this.getAttributeValue(Attributes.ATTACK_DAMAGE)));
+        boolean bl = target.hurt(DamageSource.mobAttack(this), this, null, ((int)this.getAttributeValue(Attributes.ATTACK_DAMAGE)));
         if (bl) {
             this.doEnchantDamageEffects(this, target);
         }
