From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Tue, 15 Nov 2022 00:16:14 -0600
Subject: [PATCH] Fix ender dragon part


diff --git a/src/main/java/net/minecraft/world/entity/boss/EnderDragonPart.java b/src/main/java/net/minecraft/world/entity/boss/EnderDragonPart.java
index 305a891e4b51d1031d9e9238ff00e2ea7de8d954..7a8d10e66332bf6315011921c0e9d2f0842d49cd 100644
--- a/src/main/java/net/minecraft/world/entity/boss/EnderDragonPart.java
+++ b/src/main/java/net/minecraft/world/entity/boss/EnderDragonPart.java
@@ -8,6 +8,9 @@ import net.minecraft.world.entity.EntityDimensions;
 import net.minecraft.world.entity.Pose;
 import net.minecraft.world.entity.boss.enderdragon.EnderDragon;
 
+import javax.annotation.Nullable;
+
+
 public class EnderDragonPart extends Entity {
     public final EnderDragon parentMob;
     public final String name;
@@ -39,8 +42,8 @@ public class EnderDragonPart extends Entity {
     }
 
     @Override
-    public boolean hurt(DamageSource source, float amount) {
-        return this.isInvulnerableTo(source) ? false : this.parentMob.hurt(this, source, amount);
+    public boolean hurt(DamageSource source, @Nullable Entity damagerEntity, @Nullable final org.bukkit.block.Block damagerBlock, float amount) {
+        return !this.isInvulnerableTo(source) && this.parentMob.hurt(this, damagerEntity, damagerBlock, source, amount);
     }
 
     @Override
