From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Tue, 15 Nov 2022 00:51:50 -0600
Subject: [PATCH] Multithreading entity damage step 4


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 368847dd87ae62449960633c556edae0634761b2..d3682f6cb94ed1e489f06438eeada2ab62cd33e0 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -1632,14 +1632,14 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource {
         return this.getType().fireImmune();
     }
 
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
+    public boolean causeFallDamage(float fallDistance, @Nullable Entity damagerEntity, @Nullable final org.bukkit.block.Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
         if (this.isVehicle()) {
             Iterator iterator = this.getPassengers().iterator();
 
             while (iterator.hasNext()) {
                 Entity entity = (Entity) iterator.next();
 
-                entity.causeFallDamage(fallDistance, damageMultiplier, damageSource);
+                entity.causeFallDamage(fallDistance, damagerEntity, damagerBlock, damageMultiplier, damageSource);
             }
         }
 
diff --git a/src/main/java/net/minecraft/world/entity/LivingEntity.java b/src/main/java/net/minecraft/world/entity/LivingEntity.java
index 11353995554565b7cc445331b437b61d645e3c3f..697490a39f9d8e8f5d012b79b6c7766db2fd24cf 100644
--- a/src/main/java/net/minecraft/world/entity/LivingEntity.java
+++ b/src/main/java/net/minecraft/world/entity/LivingEntity.java
@@ -116,7 +116,6 @@ import net.minecraft.world.phys.HitResult;
 import net.minecraft.world.phys.Vec3;
 import net.minecraft.world.scores.PlayerTeam;
 import org.bukkit.block.Block;
-import org.bukkit.craftbukkit.block.CraftBlock;
 import org.jetbrains.annotations.NotNull;
 import org.slf4j.Logger;
 
@@ -1935,8 +1934,8 @@ public abstract class LivingEntity extends Entity {
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
-        boolean flag = super.causeFallDamage(fallDistance, damageMultiplier, damageSource);
+    public boolean causeFallDamage(float fallDistance, final @Nullable Entity damagerEntity, final Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
+        boolean flag = super.causeFallDamage(fallDistance, damagerEntity, damagerBlock, damageMultiplier, damageSource);
         int i = this.calculateFallDamage(fallDistance, damageMultiplier);
 
         if (i > 0) {
diff --git a/src/main/java/net/minecraft/world/entity/ai/behavior/RamTarget.java b/src/main/java/net/minecraft/world/entity/ai/behavior/RamTarget.java
index 7b4ea52e18b164e5d738d3098a8df411c4023ea0..2b082044e1636fa2200bf5c2da2aedbe5e47a898 100644
--- a/src/main/java/net/minecraft/world/entity/ai/behavior/RamTarget.java
+++ b/src/main/java/net/minecraft/world/entity/ai/behavior/RamTarget.java
@@ -71,7 +71,7 @@ public class RamTarget extends Behavior<Goat> {
         Brain<?> brain = goat.getBrain();
         if (!list.isEmpty()) {
             LivingEntity livingEntity = list.get(0);
-            livingEntity.hurt(DamageSource.mobAttack(goat).setNoAggro(), block, (float)goat.getAttributeValue(Attributes.ATTACK_DAMAGE));
+            livingEntity.hurt(DamageSource.mobAttack(goat).setNoAggro(), goat, null, (float)goat.getAttributeValue(Attributes.ATTACK_DAMAGE));
             int i = goat.hasEffect(MobEffects.MOVEMENT_SPEED) ? goat.getEffect(MobEffects.MOVEMENT_SPEED).getAmplifier() + 1 : 0;
             int j = goat.hasEffect(MobEffects.MOVEMENT_SLOWDOWN) ? goat.getEffect(MobEffects.MOVEMENT_SLOWDOWN).getAmplifier() + 1 : 0;
             float f = 0.25F * (float)(i - j);
diff --git a/src/main/java/net/minecraft/world/entity/ambient/Bat.java b/src/main/java/net/minecraft/world/entity/ambient/Bat.java
index 4a0bc0676222424ad52f5b6b3f5ed40195347f74..4fa022bc7f2c91967562bded3779e48614842268 100644
--- a/src/main/java/net/minecraft/world/entity/ambient/Bat.java
+++ b/src/main/java/net/minecraft/world/entity/ambient/Bat.java
@@ -26,7 +26,7 @@ import net.minecraft.world.level.Level;
 import net.minecraft.world.level.LevelAccessor;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.phys.Vec3;
-import org.bukkit.craftbukkit.block.CraftBlock;
+import org.bukkit.block.Block;
 import org.bukkit.craftbukkit.event.CraftEventFactory; // CraftBukkit
 
 public class Bat extends AmbientCreature {
@@ -195,7 +195,7 @@ public class Bat extends AmbientCreature {
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
+    public boolean causeFallDamage(float fallDistance, final @Nullable Entity damagerEntity, final Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
         return false;
     }
 
diff --git a/src/main/java/net/minecraft/world/entity/animal/Bee.java b/src/main/java/net/minecraft/world/entity/animal/Bee.java
index 332858ea72e197f5fea1fc537dc9b7fd69f9c539..bac43df9fb8748b8c2920cb6e38c56f10e52ba3a 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Bee.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Bee.java
@@ -89,7 +89,6 @@ import net.minecraft.world.level.material.Fluid;
 import net.minecraft.world.level.pathfinder.BlockPathTypes;
 import net.minecraft.world.level.pathfinder.Path;
 import net.minecraft.world.phys.Vec3;
-import org.bukkit.craftbukkit.block.CraftBlock;
 
 
 public class Bee extends Animal implements NeutralMob, FlyingAnimal {
@@ -637,7 +636,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
+    public boolean causeFallDamage(float fallDistance, final @Nullable Entity damagerEntity, final org.bukkit.block.Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
         return false;
     }
 
diff --git a/src/main/java/net/minecraft/world/entity/animal/Cat.java b/src/main/java/net/minecraft/world/entity/animal/Cat.java
index 5aac3052d511b9c5c90a5145b9545d0eb8990e8b..32063016e50b3e8aa37f11922a4c5852d9d2c6c1 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Cat.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Cat.java
@@ -69,6 +69,8 @@ import net.minecraft.world.level.storage.loot.LootTable;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParamSets;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParams;
 import net.minecraft.world.phys.AABB;
+import org.bukkit.block.Block;
+
 
 public class Cat extends TamableAnimal {
 
@@ -232,7 +234,7 @@ public class Cat extends TamableAnimal {
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
+    public boolean causeFallDamage(float fallDistance, final @Nullable Entity damagerEntity, final Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
         return false;
     }
 
diff --git a/src/main/java/net/minecraft/world/entity/animal/Chicken.java b/src/main/java/net/minecraft/world/entity/animal/Chicken.java
index f01e508451217ee3bd7eaf9c322b2833c020514c..1965ac5b0766b42555d1095cc1152b389ddb22a9 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Chicken.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Chicken.java
@@ -28,12 +28,15 @@ import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
 import net.minecraft.world.item.crafting.Ingredient;
-import net.minecraft.world.level.ItemLike;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.gameevent.GameEvent;
 import net.minecraft.world.level.pathfinder.BlockPathTypes;
 import net.minecraft.world.phys.Vec3;
+import org.bukkit.block.Block;
+
+import javax.annotation.Nullable;
+
 
 public class Chicken extends Animal {
 
@@ -115,7 +118,7 @@ public class Chicken extends Animal {
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
+    public boolean causeFallDamage(float fallDistance, final @Nullable Entity damagerEntity, final Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
         return false;
     }
 
diff --git a/src/main/java/net/minecraft/world/entity/animal/Ocelot.java b/src/main/java/net/minecraft/world/entity/animal/Ocelot.java
index be0209df5a9243f3f9a8b25f936f6c085e1f13e0..7ca5c190d0c0b140d7201969003a2835e9ad4d35 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Ocelot.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Ocelot.java
@@ -48,6 +48,8 @@ import net.minecraft.world.level.ServerLevelAccessor;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.phys.Vec3;
+import org.bukkit.block.Block;
+
 
 public class Ocelot extends Animal {
 
@@ -139,7 +141,7 @@ public class Ocelot extends Animal {
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
+    public boolean causeFallDamage(float fallDistance, final @Nullable Entity damagerEntity, final Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
         return false;
     }
 
diff --git a/src/main/java/net/minecraft/world/entity/animal/Parrot.java b/src/main/java/net/minecraft/world/entity/animal/Parrot.java
index b2b2eab41c793b14f7fc0b409d58edc3be93600a..b50305649d44db8ee12bde89382c4dd7c5c59a23 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Parrot.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Parrot.java
@@ -65,7 +65,7 @@ import net.minecraft.world.level.block.LeavesBlock;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.pathfinder.BlockPathTypes;
 import net.minecraft.world.phys.Vec3;
-import org.bukkit.craftbukkit.block.CraftBlock;
+import org.bukkit.block.Block;
 
 
 public class Parrot extends ShoulderRidingEntity implements FlyingAnimal {
@@ -300,7 +300,7 @@ public class Parrot extends ShoulderRidingEntity implements FlyingAnimal {
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
+    public boolean causeFallDamage(float fallDistance, final @Nullable Entity damagerEntity, final Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
         return false;
     }
 
diff --git a/src/main/java/net/minecraft/world/entity/animal/Wolf.java b/src/main/java/net/minecraft/world/entity/animal/Wolf.java
index 6a2a1cb302cf3e5cb88238450451d6224df47316..510c43da24e077edde4ea5a17369d3237b5f44c0 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Wolf.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Wolf.java
@@ -314,7 +314,7 @@ public class Wolf extends TamableAnimal implements NeutralMob {
     }
 
     @Override
-    public boolean hurt(DamageSource source, final CraftBlock block, float amount) {
+    public boolean hurt(DamageSource source, @Nullable Entity damagerEntity, @Nullable final org.bukkit.block.Block damagerBlock, float amount) {
         if (this.isInvulnerableTo(source)) {
             return false;
         } else {
@@ -328,13 +328,13 @@ public class Wolf extends TamableAnimal implements NeutralMob {
                 amount = (amount + 1.0F) / 2.0F;
             }
 
-            return super.hurt(source, block, amount);
+            return super.hurt(source, damagerEntity, damagerBlock, amount);
         }
     }
 
     @Override
     public boolean doHurtTarget(Entity target) {
-        boolean flag = target.hurt(DamageSource.mobAttack(this), block, (float) ((int) this.getAttributeValue(Attributes.ATTACK_DAMAGE)));
+        boolean flag = target.hurt(DamageSource.mobAttack(this), this, null, (float) ((int) this.getAttributeValue(Attributes.ATTACK_DAMAGE)));
 
         if (flag) {
             this.doEnchantDamageEffects(this, target);
diff --git a/src/main/java/net/minecraft/world/entity/animal/allay/Allay.java b/src/main/java/net/minecraft/world/entity/animal/allay/Allay.java
index 2fac88c1c42e18ec4f8881059c8bd290aa14a3db..15e80e7d1f38c812c8a0a84f1d23931278b8b109 100644
--- a/src/main/java/net/minecraft/world/entity/animal/allay/Allay.java
+++ b/src/main/java/net/minecraft/world/entity/animal/allay/Allay.java
@@ -68,7 +68,7 @@ import net.minecraft.world.level.gameevent.PositionSource;
 import net.minecraft.world.level.gameevent.vibrations.VibrationListener;
 import net.minecraft.world.phys.AABB;
 import net.minecraft.world.phys.Vec3;
-import org.bukkit.craftbukkit.block.CraftBlock;
+import org.bukkit.block.Block;
 import org.jetbrains.annotations.Nullable;
 import org.slf4j.Logger;
 
@@ -183,7 +183,7 @@ public class Allay extends PathfinderMob implements InventoryCarrier {
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
+    public boolean causeFallDamage(float fallDistance, final @javax.annotation.Nullable Entity damagerEntity, final Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
         return false;
     }
 
diff --git a/src/main/java/net/minecraft/world/entity/animal/horse/AbstractHorse.java b/src/main/java/net/minecraft/world/entity/animal/horse/AbstractHorse.java
index 2f61a33efe86a778adbc855faa1d13e7bcb204fa..fd2e15af61be8b1606d7de22fa86d1a6e1e3cbae 100644
--- a/src/main/java/net/minecraft/world/entity/animal/horse/AbstractHorse.java
+++ b/src/main/java/net/minecraft/world/entity/animal/horse/AbstractHorse.java
@@ -71,6 +71,7 @@ import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.gameevent.GameEvent;
 import net.minecraft.world.phys.AABB;
 import net.minecraft.world.phys.Vec3;
+import org.bukkit.block.Block;
 import org.bukkit.event.entity.EntityRegainHealthEvent.RegainReason; // CraftBukkit
 
 public abstract class AbstractHorse extends Animal implements ContainerListener, HasCustomInventoryScreen, PlayerRideableJumping, Saddleable {
@@ -260,7 +261,7 @@ public abstract class AbstractHorse extends Animal implements ContainerListener,
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
+    public boolean causeFallDamage(float fallDistance, final @Nullable Entity damagerEntity, final Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
         if (fallDistance > 1.0F) {
             this.playSound(SoundEvents.HORSE_LAND, 0.4F, 1.0F);
         }
diff --git a/src/main/java/net/minecraft/world/entity/animal/horse/Llama.java b/src/main/java/net/minecraft/world/entity/animal/horse/Llama.java
index 59c8860b1311aee5d1da3a7d91ca0367660376f6..4e5ecf0eda582b2a3b02898e3b62ac5a8856b988 100644
--- a/src/main/java/net/minecraft/world/entity/animal/horse/Llama.java
+++ b/src/main/java/net/minecraft/world/entity/animal/horse/Llama.java
@@ -428,7 +428,7 @@ public class Llama extends AbstractChestedHorse implements RangedAttackMob {
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
+    public boolean causeFallDamage(float fallDistance, final @Nullable Entity damagerEntity, final org.bukkit.block.Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
         int i = this.calculateFallDamage(fallDistance, damageMultiplier);
 
         if (i <= 0) {
diff --git a/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java b/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
index 7a8c5667fe1ecd11f5deadbde783b298825004db..6a367c6c1eb25da7779f67b105a9a4aa96b7d432 100644
--- a/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
+++ b/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
@@ -56,7 +56,7 @@ import net.minecraft.util.Mth;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.phys.Vec3;
-import org.bukkit.craftbukkit.block.CraftBlock;
+import org.bukkit.block.Block;
 import org.bukkit.craftbukkit.event.CraftEventFactory;
 import org.bukkit.event.entity.EntityRegainHealthEvent;
 import org.bukkit.event.entity.EntityTargetEvent;
@@ -551,7 +551,7 @@ public class WitherBoss extends Monster implements PowerableMob, RangedAttackMob
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
+    public boolean causeFallDamage(float fallDistance, final @Nullable Entity damagerEntity, final Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
         return false;
     }
 
diff --git a/src/main/java/net/minecraft/world/entity/item/FallingBlockEntity.java b/src/main/java/net/minecraft/world/entity/item/FallingBlockEntity.java
index eb0e0f5d69cda41c28e26bf96ddb1891b3adbba7..680885de92d7a2cb991426a6be4287daa8e324d3 100644
--- a/src/main/java/net/minecraft/world/entity/item/FallingBlockEntity.java
+++ b/src/main/java/net/minecraft/world/entity/item/FallingBlockEntity.java
@@ -263,7 +263,7 @@ public class FallingBlockEntity extends Entity {
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
+    public boolean causeFallDamage(float fallDistance, final @Nullable Entity damagerEntity, final org.bukkit.block.Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
         if (!this.hurtEntities) {
             return false;
         } else {
diff --git a/src/main/java/net/minecraft/world/entity/monster/Creeper.java b/src/main/java/net/minecraft/world/entity/monster/Creeper.java
index b0fd15d8e0d454e16b34755c1d16130be7123499..4ae76a8886d465fff46d7ddf4d2679abc81659b2 100644
--- a/src/main/java/net/minecraft/world/entity/monster/Creeper.java
+++ b/src/main/java/net/minecraft/world/entity/monster/Creeper.java
@@ -40,11 +40,11 @@ import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
 import net.minecraft.world.level.Explosion;
 import net.minecraft.world.level.GameRules;
-import net.minecraft.world.level.ItemLike;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.gameevent.GameEvent;
 
 // CraftBukkit start
+import org.bukkit.block.Block;
 import org.bukkit.craftbukkit.event.CraftEventFactory;
 import org.bukkit.event.entity.CreatureSpawnEvent;
 import org.bukkit.event.entity.ExplosionPrimeEvent;
@@ -89,8 +89,8 @@ public class Creeper extends Monster implements PowerableMob {
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
-        boolean flag = super.causeFallDamage(fallDistance, damageMultiplier, damageSource);
+    public boolean causeFallDamage(float fallDistance, final @Nullable Entity damagerEntity, final Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
+        boolean flag = super.causeFallDamage(fallDistance, damagerEntity, damagerBlock, damageMultiplier, damageSource);
 
         this.swell += (int) (fallDistance * 1.5F);
         if (this.swell > this.maxSwell - 5) {
diff --git a/src/main/java/net/minecraft/world/entity/player/Player.java b/src/main/java/net/minecraft/world/entity/player/Player.java
index 61417b5f1f678bd3c74c0697c812f9ed611c1eb0..7a05e6896f261ac46fc7b7490c788c502c5e647a 100644
--- a/src/main/java/net/minecraft/world/entity/player/Player.java
+++ b/src/main/java/net/minecraft/world/entity/player/Player.java
@@ -115,7 +115,6 @@ import net.minecraft.world.phys.AABB;
 import net.minecraft.world.phys.Vec3;
 import net.minecraft.world.scores.PlayerTeam;
 import net.minecraft.world.scores.Scoreboard;
-import org.bukkit.craftbukkit.block.CraftBlock;
 import org.slf4j.Logger;
 import org.bukkit.craftbukkit.entity.CraftHumanEntity;
 import org.bukkit.craftbukkit.event.CraftEventFactory;
@@ -1776,7 +1775,7 @@ public abstract class Player extends LivingEntity {
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
+    public boolean causeFallDamage(float fallDistance, final @Nullable Entity damagerEntity, final org.bukkit.block.Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
         if (this.abilities.mayfly) {
             return false;
         } else {
@@ -1784,7 +1783,7 @@ public abstract class Player extends LivingEntity {
                 this.awardStat(Stats.FALL_ONE_CM, (int) Math.round((double) fallDistance * 100.0D));
             }
 
-            return super.causeFallDamage(fallDistance, damageMultiplier, damageSource);
+            return super.causeFallDamage(fallDistance, damagerEntity, damagerBlock, damageMultiplier, damageSource);
         }
     }
 
diff --git a/src/main/java/net/minecraft/world/entity/vehicle/Boat.java b/src/main/java/net/minecraft/world/entity/vehicle/Boat.java
index 37834a590071797b9365e67f21b679bd691b7d53..96744a11cdef85ab77d1a6377be834ddac2a1362 100644
--- a/src/main/java/net/minecraft/world/entity/vehicle/Boat.java
+++ b/src/main/java/net/minecraft/world/entity/vehicle/Boat.java
@@ -51,7 +51,6 @@ import net.minecraft.world.phys.shapes.Shapes;
 import net.minecraft.world.phys.shapes.VoxelShape;
 // CraftBukkit start
 import org.bukkit.Location;
-import org.bukkit.craftbukkit.block.CraftBlock;
 import org.bukkit.entity.Vehicle;
 import org.bukkit.event.vehicle.VehicleDamageEvent;
 import org.bukkit.event.vehicle.VehicleDestroyEvent;
@@ -883,7 +882,7 @@ public class Boat extends Entity {
                         return;
                     }
 
-                    this.causeFallDamage(this.fallDistance, 1.0F, DamageSource.FALL);
+                    this.causeFallDamage(this.fallDistance, null, null, 1.0F, DamageSource.FALL);
                     if (!this.level.isClientSide && !this.isRemoved()) {
                     // CraftBukkit start
                     Vehicle vehicle = (Vehicle) this.getBukkitEntity();
diff --git a/src/main/java/net/minecraft/world/entity/vehicle/MinecartTNT.java b/src/main/java/net/minecraft/world/entity/vehicle/MinecartTNT.java
index b0a170ffa8943e92b46397e0fca56458f1af51a8..9a92b45beea1236a696a4d7635494f8421d58176 100644
--- a/src/main/java/net/minecraft/world/entity/vehicle/MinecartTNT.java
+++ b/src/main/java/net/minecraft/world/entity/vehicle/MinecartTNT.java
@@ -18,7 +18,7 @@ import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.material.FluidState;
-import org.bukkit.craftbukkit.block.CraftBlock;
+import org.bukkit.block.Block;
 
 import javax.annotation.Nullable;
 
@@ -115,13 +115,13 @@ public class MinecartTNT extends AbstractMinecart {
     }
 
     @Override
-    public boolean causeFallDamage(float fallDistance, float damageMultiplier, DamageSource damageSource) {
+    public boolean causeFallDamage(float fallDistance, final @Nullable Entity damagerEntity, final Block damagerBlock, float damageMultiplier, DamageSource damageSource) {
         if (fallDistance >= 3.0F) {
             float f = fallDistance / 10.0F;
             this.explode(f * f);
         }
 
-        return super.causeFallDamage(fallDistance, damageMultiplier, damageSource);
+        return super.causeFallDamage(fallDistance, damagerEntity, damagerBlock, damageMultiplier, damageSource);
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/food/FoodData.java b/src/main/java/net/minecraft/world/food/FoodData.java
index 53611e72c0129887b40cc49b94ec27f93c8d937b..7445d1fcece35b29f6f22397c2e071ede20eaeaa 100644
--- a/src/main/java/net/minecraft/world/food/FoodData.java
+++ b/src/main/java/net/minecraft/world/food/FoodData.java
@@ -101,7 +101,7 @@ public class FoodData {
             ++this.tickTimer;
             if (this.tickTimer >= this.starvationRate) { // CraftBukkit - add regen rate manipulation
                 if (player.getHealth() > 10.0F || enumdifficulty == Difficulty.HARD || player.getHealth() > 1.0F && enumdifficulty == Difficulty.NORMAL) {
-                    player.hurt(DamageSource.STARVE, block, 1.0F);
+                    player.hurt(DamageSource.STARVE, null, null, 1.0F);
                 }
 
                 this.tickTimer = 0;
diff --git a/src/main/java/net/minecraft/world/level/block/BaseFireBlock.java b/src/main/java/net/minecraft/world/level/block/BaseFireBlock.java
index 65070cd458b9550889465bb17eec45eb4c7aa211..a61b4a7c2c5a2ee2d72d0265155b90510de4a2a7 100644
--- a/src/main/java/net/minecraft/world/level/block/BaseFireBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/BaseFireBlock.java
@@ -142,7 +142,7 @@ public abstract class BaseFireBlock extends Block {
             }
         }
 
-        entity.hurt(DamageSource.IN_FIRE, block, this.fireDamage);
+        entity.hurt(DamageSource.IN_FIRE, null, null, this.fireDamage);
         super.entityInside(state, world, pos, entity);
     }
 
diff --git a/src/main/java/net/minecraft/world/level/block/Block.java b/src/main/java/net/minecraft/world/level/block/Block.java
index e52328f79c5251b0c4885c8eba0f015aef8b4edf..c63f2e87ecec1758144de728ac2df1347b70d82a 100644
--- a/src/main/java/net/minecraft/world/level/block/Block.java
+++ b/src/main/java/net/minecraft/world/level/block/Block.java
@@ -459,7 +459,7 @@ public class Block extends BlockBehaviour implements ItemLike {
     }
 
     public void fallOn(Level world, BlockState state, BlockPos pos, Entity entity, float fallDistance) {
-        entity.causeFallDamage(fallDistance, 1.0F, DamageSource.FALL);
+        entity.causeFallDamage(fallDistance, null, null, 1.0F, DamageSource.FALL);
     }
 
     public void updateEntityAfterFallOn(BlockGetter world, Entity entity) {
diff --git a/src/main/java/net/minecraft/world/level/block/CampfireBlock.java b/src/main/java/net/minecraft/world/level/block/CampfireBlock.java
index 8a4007a9918a1e5631ca46d64d0ad0b5c4963b79..cce8e7bf0be6e1be2c8942395c42ccbc3986df0e 100644
--- a/src/main/java/net/minecraft/world/level/block/CampfireBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/CampfireBlock.java
@@ -94,9 +94,7 @@ public class CampfireBlock extends BaseEntityBlock implements SimpleWaterloggedB
     public void entityInside(BlockState state, Level world, BlockPos pos, Entity entity) {
         if (!new io.papermc.paper.event.entity.EntityInsideBlockEvent(entity.getBukkitEntity(), org.bukkit.craftbukkit.block.CraftBlock.at(world, pos)).callEvent()) { return; } // Paper
         if (state.getValue(CampfireBlock.LIT).booleanValue() && entity instanceof LivingEntity && !EnchantmentHelper.hasFrostWalker((LivingEntity) entity)) {
-            org.bukkit.craftbukkit.event.CraftEventFactory.blockDamage = CraftBlock.at(world, pos); // CraftBukkit
-            entity.hurt(DamageSource.IN_FIRE, block, (float) this.fireDamage);
-            org.bukkit.craftbukkit.event.CraftEventFactory.blockDamage = null; // CraftBukkit
+            entity.hurt(DamageSource.IN_FIRE, null, CraftBlock.at(world, pos), (float) this.fireDamage);
         }
 
         super.entityInside(state, world, pos, entity);
diff --git a/src/main/java/net/minecraft/world/level/block/HoneyBlock.java b/src/main/java/net/minecraft/world/level/block/HoneyBlock.java
index 683f24251baf8ef3bef8f32ba83dc7f0e8ed7d70..ea54a5e26a2e4a8b6dc3bb60b43f6a44f2467cf4 100644
--- a/src/main/java/net/minecraft/world/level/block/HoneyBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/HoneyBlock.java
@@ -47,7 +47,7 @@ public class HoneyBlock extends HalfTransparentBlock {
             world.broadcastEntityEvent(entity, (byte)54);
         }
 
-        if (entity.causeFallDamage(fallDistance, 0.2F, DamageSource.FALL)) {
+        if (entity.causeFallDamage(fallDistance, null, null, 0.2F, DamageSource.FALL)) {
             entity.playSound(this.soundType.getFallSound(), this.soundType.getVolume() * 0.5F, this.soundType.getPitch() * 0.75F);
         }
 
diff --git a/src/main/java/net/minecraft/world/level/block/MagmaBlock.java b/src/main/java/net/minecraft/world/level/block/MagmaBlock.java
index 35266867e24b11d986323dc84ff304d282bc2ed4..33112c566f5d3b22a0fbdcf77053d008fd3b8b95 100644
--- a/src/main/java/net/minecraft/world/level/block/MagmaBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/MagmaBlock.java
@@ -28,9 +28,7 @@ public class MagmaBlock extends Block {
     @Override
     public void stepOn(Level world, BlockPos pos, BlockState state, Entity entity) {
         if (!entity.isSteppingCarefully() && entity instanceof LivingEntity && !EnchantmentHelper.hasFrostWalker((LivingEntity) entity)) {
-            org.bukkit.craftbukkit.event.CraftEventFactory.blockDamage = world.getWorld().getBlockAt(pos.getX(), pos.getY(), pos.getZ()); // CraftBukkit
-            entity.hurt(DamageSource.HOT_FLOOR, block, 1.0F);
-            org.bukkit.craftbukkit.event.CraftEventFactory.blockDamage = null; // CraftBukkit
+            entity.hurt(DamageSource.HOT_FLOOR, null, world.getWorld().getBlockAt(pos.getX(), pos.getY(), pos.getZ()), 1.0F);
         }
 
         super.stepOn(world, pos, state, entity);
diff --git a/src/main/java/net/minecraft/world/level/block/PointedDripstoneBlock.java b/src/main/java/net/minecraft/world/level/block/PointedDripstoneBlock.java
index 1e0c0819f6f1936215ca2da56d42d76577ac6495..1d564869ea5b8b8759c7eb1a019bc3c036a755c4 100644
--- a/src/main/java/net/minecraft/world/level/block/PointedDripstoneBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/PointedDripstoneBlock.java
@@ -143,9 +143,7 @@ public class PointedDripstoneBlock extends Block implements Fallable, SimpleWate
     @Override
     public void fallOn(Level world, BlockState state, BlockPos pos, Entity entity, float fallDistance) {
         if (state.getValue(PointedDripstoneBlock.TIP_DIRECTION) == Direction.UP && state.getValue(PointedDripstoneBlock.THICKNESS) == DripstoneThickness.TIP) {
-            CraftEventFactory.blockDamage = CraftBlock.at(world, pos); // CraftBukkit
-            entity.causeFallDamage(fallDistance + 2.0F, 2.0F, DamageSource.STALAGMITE);
-            CraftEventFactory.blockDamage = null; // CraftBukkit
+            entity.causeFallDamage(fallDistance + 2.0F, null, CraftBlock.at(world, pos), 2.0F, DamageSource.STALAGMITE);
         } else {
             super.fallOn(world, state, pos, entity, fallDistance);
         }
diff --git a/src/main/java/net/minecraft/world/level/block/SweetBerryBushBlock.java b/src/main/java/net/minecraft/world/level/block/SweetBerryBushBlock.java
index 85355157b0684af0684e69d6df107a5fe4314ffe..7f2a1e5b8bda96ccd450429d23fbf77407a88c4d 100644
--- a/src/main/java/net/minecraft/world/level/block/SweetBerryBushBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/SweetBerryBushBlock.java
@@ -85,9 +85,7 @@ public class SweetBerryBushBlock extends BushBlock implements BonemealableBlock
                 double d1 = Math.abs(entity.getZ() - entity.zOld);
 
                 if (d0 >= 0.003000000026077032D || d1 >= 0.003000000026077032D) {
-                    CraftEventFactory.blockDamage = CraftBlock.at(world, pos); // CraftBukkit
-                    entity.hurt(DamageSource.SWEET_BERRY_BUSH, block, 1.0F);
-                    CraftEventFactory.blockDamage = null; // CraftBukkit
+                    entity.hurt(DamageSource.SWEET_BERRY_BUSH, null, CraftBlock.at(world, pos), 1.0F);
                 }
             }
 
diff --git a/src/main/java/net/minecraft/world/level/block/entity/ConduitBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
index d99978a46ea54b37cf3285679c7cc256a70de4ab..b57daf6d4077d3782855703b9aefa1c7db939f47 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
@@ -236,11 +236,9 @@ public class ConduitBlockEntity extends BlockEntity {
 
         if (blockEntity.destroyTarget != null) {
             // CraftBukkit start
-            CraftEventFactory.blockDamage = CraftBlock.at(world, pos);
-            if (blockEntity.destroyTarget.hurt(DamageSource.MAGIC, block, 4.0F)) {
+            if (blockEntity.destroyTarget.hurt(DamageSource.MAGIC, null, CraftBlock.at(world, pos), 4.0F)) {
                 world.playSound(null, blockEntity.destroyTarget.getX(), blockEntity.destroyTarget.getY(), blockEntity.destroyTarget.getZ(), SoundEvents.CONDUIT_ATTACK_TARGET, SoundSource.BLOCKS, 1.0F, 1.0F);
             }
-            CraftEventFactory.blockDamage = null;
             // CraftBukkit end
         }
 
