From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Tue, 15 Nov 2022 01:13:05 -0600
Subject: [PATCH] Fix crashes


diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index c352e40f4bf36070c4697e91d6400dc62243a997..3126c1de57ca51b2f99d482e463bc368612ae3c8 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -1672,7 +1672,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
 
         if (!messages.isEmpty()) {
             this.gameEventMessages = new AtomicReference<>(new ConcurrentLinkedDeque<>());
-            this.handleGameEventMessagesInQueue(messages.stream().toList());
+            this.handleGameEventMessagesInQueue(new ArrayList<>(messages));
         }
     }
 
