From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Tue, 15 Nov 2022 01:19:15 -0600
Subject: [PATCH] Fix crashes


diff --git a/src/main/java/net/minecraft/world/entity/monster/hoglin/HoglinBase.java b/src/main/java/net/minecraft/world/entity/monster/hoglin/HoglinBase.java
index 127948e7c8620de1f4fabce0d654fbb502510156..20ebb6f7fe06a9a3f540f90670296b916efbe889 100644
--- a/src/main/java/net/minecraft/world/entity/monster/hoglin/HoglinBase.java
+++ b/src/main/java/net/minecraft/world/entity/monster/hoglin/HoglinBase.java
@@ -19,7 +19,7 @@ public interface HoglinBase {
             g = f;
         }
 
-        boolean bl = target.hurt(DamageSource.mobAttack(attacker), g);
+        boolean bl = target.hurt(DamageSource.mobAttack(attacker), attacker, null, g);
         if (bl) {
             attacker.doEnchantDamageEffects(attacker, target);
             if (!attacker.isBaby()) {
