From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Tue, 15 Nov 2022 02:21:26 -0600
Subject: [PATCH] Fix crashes


diff --git a/src/main/java/io/papermc/paper/world/ChunkEntitySlices.java b/src/main/java/io/papermc/paper/world/ChunkEntitySlices.java
index 5afaaa1d9fbbef34a79234cb50e3b23f50cf0516..bacc1724a54ff4fee8127f47a07241ee44ee7ca4 100644
--- a/src/main/java/io/papermc/paper/world/ChunkEntitySlices.java
+++ b/src/main/java/io/papermc/paper/world/ChunkEntitySlices.java
@@ -64,7 +64,7 @@ public final class ChunkEntitySlices {
 
         this.allEntities = new EntityCollectionBySection(this);
         this.hardCollidingEntities = new EntityCollectionBySection(this);
-        this.entitiesByClass = new HashMap<>();
+        this.entitiesByClass = new ConcurrentHashMap<>();
 
         this.status = status;
     }
