From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Tue, 15 Nov 2022 04:05:49 -0600
Subject: [PATCH] Fix crash


diff --git a/src/main/java/net/minecraft/world/damagesource/CombatTracker.java b/src/main/java/net/minecraft/world/damagesource/CombatTracker.java
index e8c25e281cf7479c718fb749c92609b6b82e0a31..57b7db2799dfcfe81b18ca802cdab3a524a002e8 100644
--- a/src/main/java/net/minecraft/world/damagesource/CombatTracker.java
+++ b/src/main/java/net/minecraft/world/damagesource/CombatTracker.java
@@ -1,10 +1,5 @@
 package net.minecraft.world.damagesource;
 
-import com.google.common.collect.Lists;
-import java.util.List;
-import java.util.Optional;
-import java.util.concurrent.CopyOnWriteArrayList;
-import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
 import net.minecraft.network.chat.Component;
 import net.minecraft.tags.BlockTags;
@@ -15,6 +10,11 @@ import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.state.BlockState;
 
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+
 public class CombatTracker {
     public static final int RESET_DAMAGE_STATUS_TIME = 100;
     public static final int RESET_COMBAT_STATUS_TIME = 300;
@@ -138,7 +138,7 @@ public class CombatTracker {
             }
         }
 
-        return (LivingEntity)(player != null && g >= f / 3.0F ? player : livingEntity);
+        return player != null && g >= f / 3.0F ? player : livingEntity;
     }
 
     @Nullable
