From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Tue, 15 Nov 2022 04:08:35 -0600
Subject: [PATCH] Fix crash


diff --git a/src/main/java/io/papermc/paper/chunk/PlayerChunkLoader.java b/src/main/java/io/papermc/paper/chunk/PlayerChunkLoader.java
index 688d79a902c4ff4aec1810b4873b750848dd3ffb..1a25eb663f55483944d667b75d4ce6f4749a4fd8 100644
--- a/src/main/java/io/papermc/paper/chunk/PlayerChunkLoader.java
+++ b/src/main/java/io/papermc/paper/chunk/PlayerChunkLoader.java
@@ -24,10 +24,8 @@ import org.bukkit.craftbukkit.entity.CraftPlayer;
 import org.bukkit.entity.Player;
 import org.jetbrains.annotations.NotNull;
 
-import java.util.ArrayDeque;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.TreeSet;
+import java.util.*;
+import java.util.concurrent.CopyOnWriteArraySet;
 import java.util.concurrent.atomic.AtomicInteger;
 
 public final class PlayerChunkLoader {
@@ -79,7 +77,7 @@ public final class PlayerChunkLoader {
 
     protected final ChunkMap chunkMap;
     protected final Reference2ObjectLinkedOpenHashMap<ServerPlayer, PlayerLoaderData> playerMap = new Reference2ObjectLinkedOpenHashMap<>(512, 0.7f);
-    protected final ReferenceLinkedOpenHashSet<PlayerLoaderData> chunkSendQueue = new ReferenceLinkedOpenHashSet<>(512, 0.7f);
+    protected final Set<PlayerLoaderData> chunkSendQueue = new CopyOnWriteArraySet<>();
 
     protected final TreeSet<PlayerLoaderData> chunkLoadQueue = new TreeSet<>((final @NotNull PlayerLoaderData p1, final @NotNull PlayerLoaderData p2) -> {
         if (p1 == p2) {
@@ -559,7 +557,8 @@ public final class PlayerChunkLoader {
 
             // send chunk
 
-            final PlayerLoaderData data = this.chunkSendQueue.removeFirst();
+            final PlayerLoaderData data = this.chunkSendQueue.iterator().next();
+            this.chunkSendQueue.remove(data);
 
             final ChunkPriorityHolder queuedSend = data.sendQueue.pollFirst();
             if (queuedSend == null) {
