From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: camdenorrb <me@12oclock.dev>
Date: Fri, 18 Nov 2022 03:48:34 -0600
Subject: [PATCH] Fix crashes


diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index 5ced83a50736ecbe6b4a3bd4d6870ca7adaf4a4e..89ada7bd845dcef51a6d74d933ff25a066b730ac 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -2321,7 +2321,7 @@ public class ServerPlayer extends Player {
     }
 
     public boolean shouldFilterMessageTo(ServerPlayer player) {
-        return player == this ? false : this.textFilteringEnabled || player.textFilteringEnabled;
+        return player != this && (this.textFilteringEnabled || player.textFilteringEnabled);
     }
 
     @Override
diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftBlockStates.java b/src/main/java/org/bukkit/craftbukkit/block/CraftBlockStates.java
index b8049ee74da9a1cf7b7cf189ddb4dba5e2ca9d07..9cdebe2d5f1a2b2f5c1f08899a8191923b7ce935 100644
--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBlockStates.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBlockStates.java
@@ -51,7 +51,9 @@ public final class CraftBlockStates {
 
         @Override
         public final B createBlockState(@org.jetbrains.annotations.Nullable World world, @NotNull BlockPos blockPosition, net.minecraft.world.level.block.state.BlockState blockData, @org.jetbrains.annotations.Nullable BlockEntity tileEntity) {
-            if (tileEntity == null) {
+            if (world != null) {
+                Preconditions.checkState(tileEntity != null, "Tile is null, asynchronous access? %s", CraftBlock.at(((CraftWorld) world).getHandle(), blockPosition));
+            } else if (tileEntity == null) {
                 tileEntity = this.createTileEntity(blockPosition, blockData);
             }
             return this.createBlockState(world, (T) tileEntity);
